{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "rocky-reliance",
   "metadata": {},
   "outputs": [],
   "source": [
    "library(ggplot2)\n",
    "library(plyr)\n",
    "library(dplyr)\n",
    "library(tidyr)\n",
    "library(magrittr)\n",
    "library(lme4)\n",
    "library(lmerTest)\n",
    "library(GridLMM)\n",
    "library(snpStats)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "light-entity",
   "metadata": {},
   "outputs": [],
   "source": [
    "manhattanplot <- function(plot){\n",
    "        don <- gwasResults %>% \n",
    "\n",
    "      # Compute chromosome size\n",
    "      group_by(CHR) %>% \n",
    "      dplyr::summarise(chr_len=max(BP)) %>% \n",
    "\n",
    "      # Calculate cumulative position of each chromosome\n",
    "      mutate(tot=cumsum(chr_len)-chr_len) %>%\n",
    "      select(-chr_len) %>%\n",
    "\n",
    "      # Add this info to the initial dataset\n",
    "      left_join(gwasResults, ., by=c(\"CHR\"=\"CHR\")) %>%\n",
    "\n",
    "      # Add a cumulative position of each SNP\n",
    "      arrange(CHR, BP) %>%\n",
    "      mutate( BPcum=BP+tot)\n",
    "\n",
    "    axisdf = don %>% group_by(CHR) %>% summarize(center=( max(BPcum) + min(BPcum) ) / 2 )\n",
    "\n",
    "    return(ggplot(don, aes(x=BPcum, y=-log10(P))) +\n",
    "\n",
    "        # Show all points\n",
    "        geom_point( aes(color=as.factor(CHR)), alpha=0.8, size=1.3) +\n",
    "        scale_color_manual(values = rep(c(\"grey\", \"skyblue\"), 22 )) +\n",
    "\n",
    "        # custom X axis:\n",
    "        scale_x_continuous( label = axisdf$CHR, breaks= axisdf$center ) +\n",
    "        scale_y_continuous(expand = c(0, 0) ) +     # remove space between plot area and x axis\n",
    "        ylim(0,7) +\n",
    "\n",
    "        # Custom the theme:\n",
    "        theme_bw() +\n",
    "        theme( \n",
    "          legend.position=\"none\",\n",
    "          panel.border = element_blank(),\n",
    "          panel.grid.major.x = element_blank(),\n",
    "          panel.grid.minor.x = element_blank()\n",
    "        ))\n",
    "    }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "alien-vintage",
   "metadata": {},
   "outputs": [],
   "source": [
    "traits <- c('movement_count','mean_movement_length','speed_in_movement_mean','dist_travelled')\n",
    "\n",
    "pheno_df_ts <- read.csv(\"f1s_df_ts.csv\")\n",
    "pheno_df_36 <- read.csv(\"f1s_df_36.csv\")\n",
    "\n",
    "parents_df_ts <- read.csv(\"parents_df_ts.csv\")\n",
    "parents_df_36 <- read.csv(\"parents_df_36.csv\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "unusual-forge",
   "metadata": {},
   "outputs": [],
   "source": [
    "grm <- as.matrix(read.table(\"female_parents.biSNP.sing.HW.gatk.GQfilter.MAF05.w100s1r08.rel\"))\n",
    "colnames(grm) <- rownames(grm) <- read.table(\"female_parents.biSNP.sing.HW.gatk.GQfilter.MAF05.w100s1r08.rel.id\")[,2]\n",
    "\n",
    "f1_grm <- grm[grep(\"F1\",colnames(grm)),grep(\"F1\",colnames(grm))]\n",
    "p_grm <- grm[-grep(\"F1\",colnames(grm)),-grep(\"F1\",colnames(grm))]\n",
    "\n",
    "f1s <- grep(\"_S\",row.names(grm))\n",
    "parental_grm <- as.matrix(grm)\n",
    "\n",
    "family <- sapply(row.names(parental_grm),FUN=function(x){strsplit(x,\"_\")[[1]][4]})\n",
    "row.names(parental_grm) <- colnames(parental_grm) <- family"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "initial-stage",
   "metadata": {},
   "outputs": [],
   "source": [
    "pheno_df_ts <- pheno_df_ts[pheno_df_ts$Family %in% colnames(parental_grm),]\n",
    "\n",
    "#Filtering as needed\n",
    "combined_filtered_df <- pheno_df_ts\n",
    "combined_filtered_df = combined_filtered_df[combined_filtered_df$Empty == F,]\n",
    "combined_filtered_df = combined_filtered_df[!is.na(combined_filtered_df$Treatment),]\n",
    "combined_filtered_df = combined_filtered_df[!is.na(combined_filtered_df$Batch),]\n",
    "\n",
    "# Account for framerate\n",
    "combined_filtered_df$dist_travelled <- combined_filtered_df$dist_travelled/10\n",
    "\n",
    "#Ensure the flies still move some -- no dead flies!\n",
    "# combined_filtered_df = combined_filtered_df[which(combined_filtered_df$movement_count > 10),]\n",
    "# combined_filtered_df = combined_filtered_df[which(combined_filtered_df$movement_count < 10000),]\n",
    "# combined_filtered_df = combined_filtered_df[which(combined_filtered_df$mean_of_mean_in_movement_speed < 100),]\n",
    "combined_filtered_df = combined_filtered_df[combined_filtered_df$vel_par_in_movement_mean > 5 & combined_filtered_df$vel_par_in_movement_mean < 100,]\n",
    "# combined_filtered_df <- combined_filtered_df[which(combined_filtered_df$speed_in_movement_var < 500),]\n",
    "# combined_filtered_df <- combined_filtered_df[which(combined_filtered_df$mean_mov < 20000),]\n",
    "\n",
    "\n",
    "pheno_df_ts <- combined_filtered_df\n",
    "\n",
    "#Append meaningful individual ID\n",
    "pheno_df_ts$indiv_id <- paste0(pheno_df_ts$Batch,\"_\",pheno_df_ts$Family,\"_\",pheno_df_ts$Well_orderAsTracked)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "frank-musical",
   "metadata": {},
   "outputs": [],
   "source": [
    "pheno_df_36 <- pheno_df_ts[pheno_df_36$Family %in% colnames(parental_grm),]\n",
    "\n",
    "#Filtering as needed\n",
    "combined_filtered_df <- pheno_df_36\n",
    "combined_filtered_df = combined_filtered_df[combined_filtered_df$Empty == F,]\n",
    "combined_filtered_df = combined_filtered_df[!is.na(combined_filtered_df$Treatment),]\n",
    "combined_filtered_df = combined_filtered_df[!is.na(combined_filtered_df$Batch),]\n",
    "# Account for framerate\n",
    "combined_filtered_df$dist_travelled <- combined_filtered_df$dist_travelled/10\n",
    "\n",
    "#Ensure the flies still move some -- no dead flies!\n",
    "# combined_filtered_df = combined_filtered_df[which(combined_filtered_df$movement_count > 10),]\n",
    "# combined_filtered_df = combined_filtered_df[which(combined_filtered_df$movement_count < 10000),]\n",
    "# combined_filtered_df = combined_filtered_df[which(combined_filtered_df$mean_of_mean_in_movement_speed < 100),]\n",
    "combined_filtered_df = combined_filtered_df[combined_filtered_df$vel_par_in_movement_mean > 5 & combined_filtered_df$vel_par_in_movement_mean < 100,]\n",
    "# combined_filtered_df <- combined_filtered_df[which(combined_filtered_df$speed_in_movement_var < 500),]\n",
    "# combined_filtered_df <- combined_filtered_df[which(combined_filtered_df$mean_mov < 20000),]\n",
    "\n",
    "\n",
    "pheno_df_36 <- combined_filtered_df\n",
    "\n",
    "#Append meaningful individual ID\n",
    "pheno_df_36$indiv_id <- paste0(pheno_df_36$Batch,\"_\",pheno_df_36$Family,\"_\",pheno_df_36$Well_orderAsTracked)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "formed-collector",
   "metadata": {},
   "outputs": [],
   "source": [
    "path <- paste(\"female_parents.biSNP.sing.HW.gatk.GQfilter.MAF05.w100s1r08\", c(\".bed\", \".bim\", \".fam\"), sep = \"\")\n",
    "SNPs <- read.plink(path[1], path[2], path[3])\n",
    "\n",
    "genotype_mat <- as(SNPs$genotypes,\"numeric\")\n",
    "row.names(genotype_mat) <- sapply(row.names(genotype_mat),FUN=function(x){strsplit(x,\"_\")[[1]][4]})\n",
    "\n",
    "strs <- sapply(unique(pheno_df_ts$Family),toString)\n",
    "\n",
    "# Expand genotype matrix to fit phenotype vector length\n",
    "gt_mat <- genotype_mat[strs,]\n",
    "\n",
    "\n",
    "mean_filled_gt_mat = gt_mat\n",
    "\n",
    "fillNAX= TRUE\n",
    "if(sum(is.na(mean_filled_gt_mat))>0){\n",
    "    if(fillNAX){\n",
    "      mean_filled_gt_mat = apply(mean_filled_gt_mat,2,function(x) {\n",
    "        if(sum(is.na(x)) > 0){\n",
    "          x[is.na(x)] = mean(x,na.rm=T)\n",
    "        }\n",
    "        x\n",
    "      })\n",
    "    } else{\n",
    "      mean_filled_gt_mat[,colSums(is.na(X))>1] = 0\n",
    "    }\n",
    "  }\n",
    "\n",
    "name_split <- sapply(colnames(gt_mat),FUN=function(x){strsplit(x,split=c(\"\\\\[\"))[[1]][1]})\n",
    "chr <- sapply(name_split,FUN=function(x){strsplit(x,split=\":\")[[1]][1]})\n",
    "pos <- as.numeric(sapply(name_split,FUN=function(x){strsplit(x,split=\":\")[[1]][2]}))\n",
    "\n",
    "map_g = data.frame(snp = name_split,chr = chr, pos=pos,ref=rep(\"A\",ncol(gt_mat)),alt=rep(\"T\",ncol(gt_mat)))\n",
    "# chr <- mapvalues(map_g$chr,unique(map_g$chr),c(1,2,3,4,5,6))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "duplicate-armor",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "`summarise()` has grouped output by 'Family'. You can override using the `.groups` argument.\n",
      "\n",
      "`summarise()` has grouped output by 'Family'. You can override using the `.groups` argument.\n",
      "\n",
      "`summarise()` has grouped output by 'Family'. You can override using the `.groups` argument.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# A simple match so it is a bit easier to create the mapped data set\n",
    "Mode <- function(x) {\n",
    "  ux <- unique(x)\n",
    "  ux[which.max(tabulate(match(x, ux)))]\n",
    "}\n",
    "\n",
    "# Select slicetime and treatments of interest and get group mean (along with batch, camera, and plate)\n",
    "mapping_df_0to4 <- pheno_df_ts[pheno_df_ts$slicetime == 0 & pheno_df_ts$Treatment %in% c(\"C\",\"HS\",\"R\"),] %>% dplyr::group_by(Family,Treatment) %>% dplyr::summarize(dist_travelled = mean(dist_travelled, na.rm = TRUE),\n",
    "    mean_movement_length = mean(mean_movement_length, na.rm = TRUE),movement_count = mean(movement_count, na.rm = TRUE),\n",
    "    speed_in_movement_mean = mean(speed_in_movement_mean, na.rm = TRUE), batch = Mode(Batch), camera = Mode(Camera), plate = Mode(Plate))\n",
    "\n",
    "mapping_df_32to36 <- pheno_df_ts[pheno_df_ts$slicetime == 32 & pheno_df_ts$Treatment %in% c(\"C\",\"HS\",\"R\"),] %>% dplyr::group_by(Family,Treatment) %>% dplyr::summarize(dist_travelled = mean(dist_travelled, na.rm = TRUE),\n",
    "    mean_movement_length = mean(mean_movement_length, na.rm = TRUE),movement_count = mean(movement_count, na.rm = TRUE),\n",
    "    speed_in_movement_mean = mean(speed_in_movement_mean, na.rm = TRUE), batch = Mode(Batch), camera = Mode(Camera), plate = Mode(Plate))\n",
    "\n",
    "mapping_df_0to36 <- pheno_df_36 %>% dplyr::group_by(Family,Treatment) %>% dplyr::summarize(dist_travelled = mean(dist_travelled, na.rm = TRUE),\n",
    "    mean_movement_length = mean(mean_movement_length, na.rm = TRUE),movement_count = mean(movement_count, na.rm = TRUE),\n",
    "    speed_in_movement_mean = mean(speed_in_movement_mean, na.rm = TRUE), batch = Mode(Batch), camera = Mode(Camera), plate = Mode(Plate))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "aggregate-oxide",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "`summarise()` has grouped output by 'Family'. You can override using the `.groups` argument.\n",
      "\n",
      "`summarise()` has grouped output by 'Family'. You can override using the `.groups` argument.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Regress out fixed effects before grouping into\n",
    "pheno_df_ts_reg <- pheno_df_ts\n",
    "pheno_df_ts_reg$dist_travelled <- residuals(lm(dist_travelled ~ 1 + Batch + Camera + Plate ,pheno_df_ts))\n",
    "pheno_df_ts_reg$movement_count <- residuals(lm(movement_count ~ 1 + Batch + Camera + Plate ,pheno_df_ts))\n",
    "pheno_df_ts_reg$mean_movement_length <- residuals(lm(mean_movement_length ~ 1 + Batch + Camera + Plate ,pheno_df_ts))\n",
    "pheno_df_ts_reg$speed_in_movement_mean <- residuals(lm(speed_in_movement_mean ~ 1 + Batch + Camera + Plate ,pheno_df_ts))\n",
    "\n",
    "# Select slicetime and treatments of interest and get group mean (along with batch, camera, and plate)\n",
    "mapping_df_0to4_regressed <- pheno_df_ts_reg[pheno_df_ts$slicetime == 0 & pheno_df_ts$Treatment %in% c(\"C\",\"HS\",\"R\"),] %>% \n",
    "    dplyr::group_by(Family,Treatment) %>% \n",
    "    dplyr::summarize(dist_travelled = mean(dist_travelled, na.rm = TRUE),\n",
    "                     mean_movement_length = mean(mean_movement_length, na.rm = TRUE),\n",
    "                     movement_count = mean(movement_count, na.rm = TRUE),\n",
    "                     speed_in_movement_mean = mean(speed_in_movement_mean, na.rm = TRUE), \n",
    "                     batch = Mode(Batch), camera = Mode(Camera), plate = Mode(Plate))\n",
    "\n",
    "mapping_df_32to36_regressed <- pheno_df_ts_reg[pheno_df_ts$slicetime == 32 & pheno_df_ts$Treatment %in% c(\"C\",\"HS\",\"R\"),] %>% \n",
    "    dplyr::group_by(Family,Treatment) %>% \n",
    "    dplyr::summarize(dist_travelled = mean(dist_travelled, na.rm = TRUE),\n",
    "                     mean_movement_length = mean(mean_movement_length, na.rm = TRUE),\n",
    "                     movement_count = mean(movement_count, na.rm = TRUE),\n",
    "                     speed_in_movement_mean = mean(speed_in_movement_mean, na.rm = TRUE), \n",
    "                     batch = Mode(Batch), camera = Mode(Camera), plate = Mode(Plate))\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "forward-tension",
   "metadata": {},
   "outputs": [],
   "source": [
    "# null_model = GridLMM_ML(paste(trait, '1 + Treatment + (1 + Treatment | Family)',\n",
    "#                                                      sep = ' ~ '),mapping_df,relmat = list(Family = parental_grm),tolerance = 1e-3,save_V_folder = 'V_folder',)\n",
    "# RE_setup = null_model$setup$RE_setup\n",
    "# full_RE_names = names(RE_setup)\n",
    "# names(RE_setup) = c('G','GxE_HS','GxE_R')\n",
    "\n",
    "# null_model$results\n",
    "\n",
    "# h2_start <- null_model$results[,c('Family..Intercept..REML','Family.TreatmentHS.REML','Family.TreatmentR.REML')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "confirmed-hearing",
   "metadata": {},
   "outputs": [],
   "source": [
    "# out_0to4_grm = list()\n",
    "# for (i in 1:4){\n",
    "#     print(i)\n",
    "#     trait = traits[i]\n",
    "#     print(trait)\n",
    "#     mother_mean_mod <- GridLMM_GWAS(as.formula(paste(trait, '1 + batch + plate + camera + Treatment + (1 + Treatment | Family)',\n",
    "#                                                      sep = ' ~ ')),~1 + Treatment,~1,data=mapping_df_0to4, X = mean_filled_gt_mat,\n",
    "#                                                      X_ID = 'Family', relmat = list(Family = parental_grm))\n",
    "#     out_0to4_grm[[i]] <- mother_mean_mod\n",
    "# }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "cheap-remainder",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 1\n",
      "[1] \"movement_count\"\n",
      "[1] \"Estimating h2_start via null model\"\n",
      "[1] \"step_size: 0.5, num_to_test: 4\"\n",
      "[1] \"step_size: 0.25, num_to_test: 7\"\n",
      "[1] \"step_size: 0.125, num_to_test: 7\"\n",
      "[1] \"step_size: 0.125, num_to_test: 3\"\n",
      "[1] \"step_size: 0.0625, num_to_test: 11\"\n",
      "[1] \"step_size: 0.0625, num_to_test: 8\"\n",
      "[1] \"step_size: 0.03125, num_to_test: 26\"\n",
      "[1] \"step_size: 0.03125, num_to_test: 2\"\n",
      "[1] \"step_size: 0.015625, num_to_test: 17\"\n",
      "[1] \"step_size: 0.015625, num_to_test: 5\"\n",
      "[1] \"step_size: 0.0078125, num_to_test: 17\"\n",
      "[1] \"step_size: 0.00390625, num_to_test: 17\"\n",
      "[1] \"step_size: 0.00390625, num_to_test: 5\"\n",
      "[1] \"step_size: 0.001953125, num_to_test: 28\"\n",
      "[1] \"GridLMM_posterior h2s:\"\n",
      "     Family..Intercept. Family.TreatmentHS Family.TreatmentR\n",
      "[1,]           0.671875         0.02734375                 0\n",
      "[1] \"step: 1, num_h2s: 1, num active: 565448\"\n",
      "[1] \"step: 2, num_h2s: 17, num active: 565448\"\n",
      "[1] \"step: 3, num_h2s: 50, num active: 121763\"\n",
      "[1] \"step: 4, num_h2s: 37, num active: 8527\"\n",
      "[1] \"step: 5, num_h2s: 25, num active: 529\"\n",
      "[1] \"step: 6, num_h2s: 13, num active: 60\"\n",
      "[1] \"step: 7, num_h2s: 8, num active: 8\"\n",
      "[1] \"step: 8, num_h2s: 6, num active: 1\"\n",
      "[1] 2\n",
      "[1] \"mean_movement_length\"\n",
      "[1] \"Estimating h2_start via null model\"\n",
      "[1] \"step_size: 0.5, num_to_test: 4\"\n",
      "[1] \"step_size: 0.25, num_to_test: 7\"\n",
      "[1] \"step_size: 0.125, num_to_test: 11\"\n",
      "[1] \"step_size: 0.125, num_to_test: 3\"\n",
      "[1] \"step_size: 0.0625, num_to_test: 11\"\n",
      "[1] \"step_size: 0.0625, num_to_test: 7\"\n",
      "[1] \"step_size: 0.03125, num_to_test: 17\"\n",
      "[1] \"step_size: 0.03125, num_to_test: 5\"\n",
      "[1] \"step_size: 0.015625, num_to_test: 28\"\n",
      "[1] \"step_size: 0.015625, num_to_test: 4\"\n",
      "[1] \"step_size: 0.0078125, num_to_test: 28\"\n",
      "[1] \"step_size: 0.00390625, num_to_test: 17\"\n",
      "[1] \"step_size: 0.00390625, num_to_test: 5\"\n",
      "[1] \"step_size: 0.001953125, num_to_test: 28\"\n",
      "[1] \"step_size: 0.001953125, num_to_test: 9\"\n",
      "[1] \"GridLMM_posterior h2s:\"\n",
      "     Family..Intercept. Family.TreatmentHS Family.TreatmentR\n",
      "[1,]          0.4277344         0.04492188                 0\n",
      "[1] \"step: 1, num_h2s: 1, num active: 565448\"\n",
      "[1] \"step: 2, num_h2s: 17, num active: 565448\"\n",
      "[1] \"step: 3, num_h2s: 32, num active: 220869\"\n",
      "[1] \"step: 4, num_h2s: 48, num active: 25001\"\n",
      "[1] \"step: 5, num_h2s: 60, num active: 3753\"\n",
      "[1] \"step: 6, num_h2s: 16, num active: 586\"\n",
      "[1] \"step: 7, num_h2s: 22, num active: 57\"\n",
      "[1] \"step: 8, num_h2s: 12, num active: 6\"\n",
      "[1] \"step: 9, num_h2s: 6, num active: 1\"\n",
      "[1] 3\n",
      "[1] \"speed_in_movement_mean\"\n",
      "[1] \"Estimating h2_start via null model\"\n",
      "[1] \"step_size: 0.5, num_to_test: 4\"\n",
      "[1] \"step_size: 0.25, num_to_test: 7\"\n",
      "[1] \"step_size: 0.125, num_to_test: 7\"\n",
      "[1] \"step_size: 0.125, num_to_test: 3\"\n",
      "[1] \"step_size: 0.0625, num_to_test: 11\"\n",
      "[1] \"step_size: 0.0625, num_to_test: 3\"\n",
      "[1] \"step_size: 0.03125, num_to_test: 11\"\n",
      "[1] \"step_size: 0.015625, num_to_test: 11\"\n",
      "[1] \"step_size: 0.015625, num_to_test: 3\"\n",
      "[1] \"step_size: 0.0078125, num_to_test: 11\"\n",
      "[1] \"step_size: 0.0078125, num_to_test: 3\"\n",
      "[1] \"step_size: 0.00390625, num_to_test: 18\"\n",
      "[1] \"step_size: 0.001953125, num_to_test: 11\"\n",
      "[1] \"step_size: 0.001953125, num_to_test: 3\"\n",
      "[1] \"GridLMM_posterior h2s:\"\n",
      "     Family..Intercept. Family.TreatmentHS Family.TreatmentR\n",
      "[1,]          0.6757812                  0                 0\n",
      "[1] \"step: 1, num_h2s: 1, num active: 565448\"\n",
      "[1] \"step: 2, num_h2s: 11, num active: 565448\"\n",
      "[1] \"step: 3, num_h2s: 26, num active: 80693\"\n",
      "[1] \"step: 4, num_h2s: 14, num active: 3794\"\n",
      "[1] \"step: 5, num_h2s: 10, num active: 374\"\n",
      "[1] \"step: 6, num_h2s: 4, num active: 37\"\n",
      "[1] \"step: 7, num_h2s: 4, num active: 4\"\n",
      "[1] \"step: 8, num_h2s: 4, num active: 1\"\n",
      "[1] 4\n",
      "[1] \"dist_travelled\"\n",
      "[1] \"Estimating h2_start via null model\"\n",
      "[1] \"step_size: 0.5, num_to_test: 4\"\n",
      "[1] \"step_size: 0.25, num_to_test: 7\"\n",
      "[1] \"step_size: 0.125, num_to_test: 7\"\n",
      "[1] \"step_size: 0.125, num_to_test: 3\"\n",
      "[1] \"step_size: 0.0625, num_to_test: 11\"\n",
      "[1] \"step_size: 0.0625, num_to_test: 3\"\n",
      "[1] \"step_size: 0.03125, num_to_test: 11\"\n",
      "[1] \"step_size: 0.03125, num_to_test: 3\"\n",
      "[1] \"step_size: 0.015625, num_to_test: 11\"\n",
      "[1] \"step_size: 0.015625, num_to_test: 3\"\n",
      "[1] \"step_size: 0.0078125, num_to_test: 11\"\n",
      "[1] \"step_size: 0.0078125, num_to_test: 3\"\n",
      "[1] \"step_size: 0.00390625, num_to_test: 11\"\n",
      "[1] \"step_size: 0.00390625, num_to_test: 3\"\n",
      "[1] \"step_size: 0.001953125, num_to_test: 18\"\n",
      "[1] \"GridLMM_posterior h2s:\"\n",
      "     Family..Intercept. Family.TreatmentHS Family.TreatmentR\n",
      "[1,]          0.6660156                  0                 0\n",
      "[1] \"step: 1, num_h2s: 1, num active: 565448\"\n",
      "[1] \"step: 2, num_h2s: 11, num active: 565448\"\n",
      "[1] \"step: 3, num_h2s: 18, num active: 81526\"\n",
      "[1] \"step: 4, num_h2s: 16, num active: 4298\"\n",
      "[1] \"step: 5, num_h2s: 8, num active: 347\"\n",
      "[1] \"step: 6, num_h2s: 4, num active: 35\"\n",
      "[1] \"step: 7, num_h2s: 4, num active: 5\"\n"
     ]
    }
   ],
   "source": [
    "out_0to4_grm_regressed = list()\n",
    "for (i in 1:4){\n",
    "    print(i)\n",
    "    trait = traits[i]\n",
    "    print(trait)\n",
    "    trait_formula <- as.formula(paste(trait, '1 + Treatment + (1 + Treatment | Family)', sep = ' ~ '))\n",
    "    mother_mean_mod <- GridLMM_GWAS(trait_formula,\n",
    "                                    ~1 + Treatment,\n",
    "                                    ~1,\n",
    "                                    data=mapping_df_0to4_regressed, \n",
    "                                    X = mean_filled_gt_mat,\n",
    "                                    X_ID = 'Family', \n",
    "                                    relmat = list(Family = parental_grm))\n",
    "     out_0to4_grm_regressed[[i]] <- mother_mean_mod\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "thorough-romance",
   "metadata": {},
   "outputs": [],
   "source": [
    "# # Map using GridLMM -- parental grm covariance matrix for family effect\n",
    "# # https://github.com/deruncie/GridLMM/blob/master/vignettes/Running_GridLMM_GWAS.Rmd for an example\n",
    "\n",
    "# out_32to36_grm = list()\n",
    "# for (i in 1:4){\n",
    "#     print(i)\n",
    "#     trait = traits[i]\n",
    "#     print(trait)\n",
    "#     mother_mean_mod <- GridLMM_GWAS(as.formula(paste(trait, '1 + batch + plate + camera + Treatment + (1 + Treatment | Family)',\n",
    "#                                                      sep = ' ~ ')),~1 + Treatment,~1,data=mapping_df_32to36, X = mean_filled_gt_mat,\n",
    "#                                                      X_ID = 'Family', relmat = list(Family = parental_grm))\n",
    "#     out_32to36_grm[[i]] <- mother_mean_mod\n",
    "# }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "under-valley",
   "metadata": {},
   "outputs": [],
   "source": [
    "out_32to36_grm_regressed = list()\n",
    "for (i in 1:4){\n",
    "    trait = traits[i]\n",
    "    print(trait)\n",
    "    trait_formula <- as.formula(paste(trait, '1 + Treatment + (1 + Treatment | Family)', sep = ' ~ '))\n",
    "    mother_mean_mod <- GridLMM_GWAS(trait_formula,\n",
    "                                    ~1 + Treatment,\n",
    "                                    ~1,\n",
    "                                    data = mapping_df_32to36_regressed,\n",
    "                                    X = mean_filled_gt_mat,\n",
    "                                    X_ID = 'Family', \n",
    "                                    relmat = list(Family = parental_grm),\n",
    "                                    algorithm=\"Fast\")\n",
    "    out_32to36_grm_regressed[[i]] <- mother_mean_mod\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "89f1fe81",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "out_32to36_grm_regressed[[1]][[1]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "ambient-credit",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"done\"\n"
     ]
    }
   ],
   "source": [
    "print(\"done\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "boring-dayton",
   "metadata": {},
   "outputs": [],
   "source": [
    "# # Map using GridLMM -- parental grm covariance matrix for family effect\n",
    "# # https://github.com/deruncie/GridLMM/blob/master/vignettes/Running_GridLMM_GWAS.Rmd for an example\n",
    "\n",
    "# out_36_grm = list()\n",
    "# for (i in 1:4){\n",
    "#     print(i)\n",
    "#     trait = traits[i]\n",
    "#     print(trait)\n",
    "#     mother_mean_mod <- GridLMM_GWAS(as.formula(paste(trait, '1 + batch + plate + camera + Treatment + (1 + Treatment | Family)',\n",
    "#                                                      sep = ' ~ ')),~1 + Treatment,~1,data=mapping_df_36, X = mean_filled_gt_mat,\n",
    "#                                                      X_ID = 'Family', relmat = list(Family = parental_grm))\n",
    "#     out_36_grm[[i]] <- mother_mean_mod\n",
    "# }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "collect-cabin",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"movement_count\"\n"
     ]
    },
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'out_32_grm' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'out_32_grm' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "i = 1\n",
    "print(traits[i])\n",
    "output = out_32_grm[[i]] # Trait 1 = traits[1]\n",
    "trait = traits[i]\n",
    "\n",
    "as.data.frame(output)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "efficient-rubber",
   "metadata": {},
   "outputs": [],
   "source": [
    "i = 1\n",
    "print(traits[i])\n",
    "output = out_0to36_grm[[i]] # Trait 1 = traits[1]\n",
    "trait = traits[i]\n",
    "\n",
    "qq::qq_plot(output$results$p_value_REML.1) + ggtitle(paste0(\"GridLMM: Control - Main Effect - \", traits[i]))\n",
    "ggsave(paste0(\"plots/0to36_GRM_qq_Ctrl_\",trait,\".png\"))\n",
    "gwasResults <- data.frame(BP = map_g$pos[1:length(output$results$p_value_REML.1)], CHR=chr[1:length(output$results$p_value_REML.1)], P = output$results$p_value_REML.1)\n",
    "gwasResults <- gwasResults[!is.na(gwasResults$P),]\n",
    "manhattanplot(gwasResults) + ggtitle(paste0(\"GridLMM: Control - Main Effect - \", traits[i]))\n",
    "ggsave(paste0(\"plots/0to36_GRM_manh_Ctrl_\",trait,\".png\"))\n",
    "\n",
    "qq::qq_plot(output$results$p_value_REML.2) + ggtitle(paste0(\"GridLMM: GxE Effect - HS - \", traits[i]))\n",
    "ggsave(paste0(\"plots/0to36_GRM_qq_GxE_HS_\",trait,\".png\"))\n",
    "gwasResults <- data.frame(BP = map_g$pos[1:length(output$results$p_value_REML.2)], CHR=chr[1:length(output$results$p_value_REML.2)], P = output$results$p_value_REML.2)\n",
    "gwasResults <- gwasResults[!is.na(gwasResults$P),]\n",
    "manhattanplot(gwasResults) + ggtitle(paste0(\"GridLMM: GxE Effect - HS - \", traits[i]))\n",
    "ggsave(paste0(\"plots/0to36_GRM_manh_GxE_HS_\",trait,\".png\"))\n",
    "\n",
    "qq::qq_plot(output$results$p_value_REML.3) + ggtitle(paste0(\"GridLMM: GxE Effect - R - \", traits[i]))\n",
    "ggsave(paste0(\"plots/0to36_GRM_qq_GxE_R_\",trait,\".png\"))\n",
    "gwasResults <- data.frame(BP = map_g$pos[1:length(output$results$p_value_REML.3)], CHR=chr[1:length(output$results$p_value_REML.3)], P = output$results$p_value_REML.3)\n",
    "gwasResults <- gwasResults[!is.na(gwasResults$P),]\n",
    "manhattanplot(gwasResults) + ggtitle(paste0(\"GridLMM: GxE Effect - R - \", traits[i]))\n",
    "ggsave(paste0(\"plots/0to36_GRM_manh_GxE_R_\",trait,\".png\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "steady-biodiversity",
   "metadata": {},
   "outputs": [],
   "source": [
    "i = 2\n",
    "print(traits[i])\n",
    "output = out_0to4_grm[[i]] # Trait 1 = traits[1]\n",
    "trait = traits[i]\n",
    "\n",
    "qq::qq_plot(output$results$p_value_REML.1) + ggtitle(paste0(\"GridLMM: Control - Main Effect - \", traits[i]))\n",
    "ggsave(paste0(\"plots/T0_GRM_qq_Ctrl_\",trait,\".png\"))\n",
    "gwasResults <- data.frame(BP = map_g$pos[1:length(output$results$p_value_REML.1)], CHR=chr[1:length(output$results$p_value_REML.1)], P = output$results$p_value_REML.1)\n",
    "gwasResults <- gwasResults[!is.na(gwasResults$P),]\n",
    "manhattanplot(gwasResults) + ggtitle(paste0(\"GridLMM: Control - Main Effect - \", traits[i]))\n",
    "ggsave(paste0(\"plots/T0_GRM_manh_Ctrl_\",trait,\".png\"))\n",
    "\n",
    "qq::qq_plot(output$results$p_value_REML.2) + ggtitle(paste0(\"GridLMM: GxE Effect - HS - \", traits[i]))\n",
    "ggsave(paste0(\"plots/T0_GRM_qq_GxE_HS_\",trait,\".png\"))\n",
    "gwasResults <- data.frame(BP = map_g$pos[1:length(output$results$p_value_REML.2)], CHR=chr[1:length(output$results$p_value_REML.2)], P = output$results$p_value_REML.2)\n",
    "gwasResults <- gwasResults[!is.na(gwasResults$P),]\n",
    "manhattanplot(gwasResults) + ggtitle(paste0(\"GridLMM: GxE Effect - HS - \", traits[i]))\n",
    "ggsave(paste0(\"plots/T0_GRM_manh_GxE_HS_\",trait,\".png\"))\n",
    "\n",
    "qq::qq_plot(output$results$p_value_REML.3) + ggtitle(paste0(\"GridLMM: GxE Effect - R - \", traits[i]))\n",
    "ggsave(paste0(\"plots/T0_GRM_qq_GxE_R_\",trait,\".png\"))\n",
    "gwasResults <- data.frame(BP = map_g$pos[1:length(output$results$p_value_REML.3)], CHR=chr[1:length(output$results$p_value_REML.3)], P = output$results$p_value_REML.3)\n",
    "gwasResults <- gwasResults[!is.na(gwasResults$P),]\n",
    "manhattanplot(gwasResults) + ggtitle(paste0(\"GridLMM: GxE Effect - R - \", traits[i]))\n",
    "ggsave(paste0(\"plots/T0_GRM_manh_GxE_R_\",trait,\".png\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "statutory-symphony",
   "metadata": {},
   "outputs": [],
   "source": [
    "i = 1\n",
    "print(traits[i])\n",
    "output = out_32to36_grm[[i]] # Trait 1 = traits[1]\n",
    "trait = traits[i]\n",
    "\n",
    "qq::qq_plot(output$results$p_value_REML.1) + ggtitle(paste0(\"GridLMM: Control - Main Effect - \", traits[i]))\n",
    "ggsave(paste0(\"plots/T32_GRM_qq_Ctrl_\",trait,\".png\"))\n",
    "gwasResults <- data.frame(BP = map_g$pos[1:length(output$results$p_value_REML.1)], CHR=chr[1:length(output$results$p_value_REML.1)], P = output$results$p_value_REML.1)\n",
    "gwasResults <- gwasResults[!is.na(gwasResults$P),]\n",
    "manhattanplot(gwasResults) + ggtitle(paste0(\"GridLMM: Control - Main Effect - \", traits[i]))\n",
    "ggsave(paste0(\"plots/T32_GRM_manh_Ctrl_\",trait,\".png\"))\n",
    "\n",
    "qq::qq_plot(output$results$p_value_REML.2) + ggtitle(paste0(\"GridLMM: GxE Effect - HS - \", traits[i]))\n",
    "ggsave(paste0(\"plots/T32_GRM_qq_GxE_HS_\",trait,\".png\"))\n",
    "gwasResults <- data.frame(BP = map_g$pos[1:length(output$results$p_value_REML.2)], CHR=chr[1:length(output$results$p_value_REML.2)], P = output$results$p_value_REML.2)\n",
    "gwasResults <- gwasResults[!is.na(gwasResults$P),]\n",
    "manhattanplot(gwasResults) + ggtitle(paste0(\"GridLMM: GxE Effect - HS - \", traits[i]))\n",
    "ggsave(paste0(\"plots/T32_GRM_manh_GxE_HS_\",trait,\".png\"))\n",
    "\n",
    "qq::qq_plot(output$results$p_value_REML.3) + ggtitle(paste0(\"GridLMM: GxE Effect - R - \", traits[i]))\n",
    "ggsave(paste0(\"plots/T32_GRM_qq_GxE_R_\",trait,\".png\"))\n",
    "gwasResults <- data.frame(BP = map_g$pos[1:length(output$results$p_value_REML.3)], CHR=chr[1:length(output$results$p_value_REML.3)], P = output$results$p_value_REML.3)\n",
    "gwasResults <- gwasResults[!is.na(gwasResults$P),]\n",
    "manhattanplot(gwasResults) + ggtitle(paste0(\"GridLMM: GxE Effect - R - \", traits[i]))\n",
    "ggsave(paste0(\"plots/T32_GRM_manh_GxE_R_\",trait,\".png\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "scenic-simpson",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"movement_count\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“`data_frame()` was deprecated in tibble 1.1.0.\n",
      "Please use `tibble()` instead.\n",
      "\u001b[90mThis warning is displayed once every 8 hours.\u001b[39m\n",
      "\u001b[90mCall `lifecycle::last_warnings()` to see where this warning was generated.\u001b[39m”\n",
      "Warning message:\n",
      "“Removed 1 rows containing missing values (geom_abline).”\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n"
     ]
    },
    {
     "ename": "ERROR",
     "evalue": "Error in grid.newpage(): could not open file 'plots/T32_GRM_qq_Ctrl_movement_count.png'\n",
     "output_type": "error",
     "traceback": [
      "Error in grid.newpage(): could not open file 'plots/T32_GRM_qq_Ctrl_movement_count.png'\nTraceback:\n",
      "1. ggsave(paste0(\"plots/T32_GRM_qq_Ctrl_\", trait, \".png\"))",
      "2. grid.draw(plot)",
      "3. grid.draw.ggplot(plot)",
      "4. print(x)",
      "5. print.ggplot(x)",
      "6. grid.newpage()"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA0gAAANICAMAAADKOT/pAAADAFBMVEUAAAABAQECAgIDAwME\nBAQFBQUGBgYHBwcICAgJCQkKCgoLCwsMDAwNDQ0ODg4PDw8QEBARERESEhITExMUFBQVFRUW\nFhYXFxcYGBgZGRkaGhobGxscHBwdHR0eHh4fHx8gICAhISEiIiIjIyMkJCQlJSUmJiYnJyco\nKCgpKSkqKiorKyssLCwtLS0uLi4vLy8wMDAxMTEyMjIzMzM0NDQ1NTU2NjY3Nzc4ODg5OTk6\nOjo7Ozs8PDw9PT0+Pj4/Pz9AQEBBQUFCQkJDQ0NERERFRUVGRkZHR0dISEhJSUlKSkpLS0tM\nTExNTU1OTk5PT09QUFBRUVFSUlJTU1NUVFRVVVVWVlZXV1dYWFhZWVlaWlpbW1tcXFxdXV1e\nXl5fX19gYGBhYWFiYmJjY2NkZGRlZWVmZmZnZ2doaGhpaWlqampra2tsbGxtbW1ubm5vb29w\ncHBxcXFycnJzc3N0dHR1dXV2dnZ3d3d4eHh5eXl6enp7e3t8fHx9fX1+fn5/f3+AgICBgYGC\ngoKDg4OEhISFhYWGhoaHh4eIiIiJiYmKioqLi4uMjIyNjY2Ojo6Pj4+QkJCRkZGSkpKTk5OU\nlJSVlZWWlpaXl5eYmJiZmZmampqbm5ucnJydnZ2enp6fn5+goKChoaGioqKjo6OkpKSlpaWm\npqanp6eoqKipqamqqqqrq6usrKytra2urq6vr6+wsLCxsbGysrKzs7O0tLS1tbW2tra3t7e4\nuLi5ubm6urq7u7u8vLy9vb2+vr6/v7/AwMDBwcHCwsLDw8PExMTFxcXGxsbHx8fIyMjJycnK\nysrLy8vMzMzNzc3Ozs7Pz8/Q0NDR0dHS0tLT09PU1NTV1dXW1tbX19fY2NjZ2dna2trb29vc\n3Nzd3d3e3t7f39/g4ODh4eHi4uLj4+Pk5OTl5eXm5ubn5+fo6Ojp6enq6urr6+vs7Ozt7e3u\n7u7v7+/w8PDx8fHy8vLz8/P09PT19fX29vb39/f4+Pj5+fn6+vr7+/v8/Pz9/f3+/v7////i\nsF19AAAACXBIWXMAABJ0AAASdAHeZh94AAAgAElEQVR4nO3dCXwTZf7H8ac3UGgRD8Byubi6\nggpaVFhR8Ebd8vdAWBFdFFDwvo8FEa9F110X110UL3QXFRUV14vLA/FCEFEuFRFQkKMgVym9\nn38mSfs8k2aeSTLze2YmfD+vl22aTuY3pX2bZJJMGEcIOY55vQEIpUOAhJALARJCLgRICLkQ\nICHkQoCEkAsBEkIuBEgIuVBqkOpeOqdrs/Z9x22Vz7yevSW+aNmO8w9Zj4avpzB2Yf3pGYyd\nGu+cmHaO77Vvk9/1n6Xckqpv18X/xhg2Ne756rGmn0FUwURbQl8vPbHZIM7nH9v02qS3izjH\nc73a8KCXEqQlPRjLOng/xpr/VzrXDlL+7ujpYfWQYs4x93lrxgp/k83YebWKTfmBnRP/GwpI\nirHWkLp1j7Yt9HV3dvzDnLdnpzyd9HYR53iuVxse9FKBtHhf1uuTPZz/fGtG1nRx9pLpG8UX\njSHlsZcjJ6tb5UUgxZ5j6rsCNui7Ol79Qnt2u2JbUoCkGmv6GUQhSDXSl1UZnUMfN7JjU9gu\n4gDJo1KAVHU4uy56JTGJda6LnrulxrRQY0glGedHTs5g/SOQYs8xdQy7I3JiaU7utsYbUVMZ\n+Sz93s1bYA1JOTZ2/ZFiIFWEf7B1rJ/FhbnN32NFtfX3UrxQnH+P5HK8gr28FCA9w35bET1Z\nV9JzCef3sYXzu2duvz18s6hizHEFPUeXNYZ0da8mO8MnL2P/jUCKPYePZH+oX34mK9oTPXkW\nmxxeQ7/Wbfv9xzj1AJv3UCu2z0kfcv4H417L1fVbIC9kDanR2OqHTjygRZfbNodOhn8Gaf3R\nzJCuN4b2+6PxcWjoy48v+E2L4n9F/g6fPK3VIRd91LBdcXqAzR6Wm9H1urKq247K7/okN/1o\nl7EJ4TNuZuNMK453Ifm7sf8e5uo3yvSPeDUL/3gfs5H2K0D2pQDpDBZzz+A+Nm3/dqftDv8R\nbu3Bso9uz3rmN4Y0gU0xTlW1OnJhFFLMOTKkkeyB+pPV4f8XD2XZ3btnsyHc+L1fwA4b1JXl\nfMlfuJb9btw79VsgL6SAFDO28hhWeOKJhaz7ngZIDeuPZoY0Yww7cNyUaTeyg8eFbtk+nJV1\nZM+m7JTQHa/aP7K8XkewjOfrtytOD7CumSed24z94aSW5/RmbLrpR5vJ+oQX6sR+MK04zoVM\n343995ATG2X6RzRDUq0AJVIKkPZnX5rPuI+1GlcT/SO8gfVYz/nzuawxpPWZJcapd9l99ZBi\nzuHfvPVF/fLdYu71v8o6f8v5is5smvF7Z/eHrg2HsWvrb4lEt0BeSAEpZuxz7MQyzst6sbkN\nkMT6Iylu2i3N6vhV6P7SScYt0cns2E2cv5G1727rW0gPsJz3OP8ikx0S+ne6g11k2urqfbNK\nQ8vMZ73NK258IfN3Y/895KSNkv99zJBUK0CJlDyk6gxWZnze0iPcX4w/4zAY449wa17OGuP0\nNXEg8T7hOzuXse/qIcWcI9eaLTd93Y3NNj7NYN2N33v4Xv5C4/qrHlKP2IUUkGLGPj8gfLNn\nPHu2AZJYfyRp93dfboZ0Hgvvn1/fpGUt75ixwjh9LvtIBely49Mh7LXQx+XsBPNWjwhf29/E\nnjSvuPGFzN+N/feQkzZKnmSGpFoBSqTkIdVlsdXG542RP6yRxp/xncYZxh/hPBb+nz3/Oh6k\nicadnapW3XkDJPM5cvuxFfKXVVltIyfaZFeHfu93GSfXyJDujF1IBSnO2Lqv+kiQxPojSbu/\nh3MzpKLCyI6XY9iKX9gx4ZM71pWrID1hfOrJ1of/DXubt3pOeGbHJttNK258oZjvxv57SEkb\nZZpkhqRYAUqoFG7aHcjebjj9QQRS+Nds/BFOZjeFzy+PB2lT1hnGDaq/CEjmc+S6sDfrT9Zs\n38G/N/54jHqxVaHfe/hOmgmSsQWmhSRIYfAfW24I3/TMtaG7SEyCJNYfyfqm3S5xXfXpR2xg\nwyLy36Np/gPGGMNEyErYhGmra/Zvsot/zgaZV9z4QjHfjf33kJI2yjTJDEmxApRQKUAawO5p\nOP1IBFL4zrvxRzglCqkuNw4kfmp2Kb+UrRSQzOfIXRL6M4/2BDuZf1f/N3BC+P/A4T8rEyRj\nC0wLSZBGGv1guSHvt2D7DJm44H4Jklh/JGtI21mbcdFWzTLuvEST/x5N8xuZMG01H8le4Tey\nd8wrbnyheN+N70DaKNOkKKR5EUiKFaCESgHSS6xdef3p02IgfRq9afdTvGsk/iSbVLXP0VyC\nZDpHbhprW//8gzND94QrM4siXxyYVWkJybSQ6qadeexvM54zdgs+lBIkvl/bhrNXs+PDn1dO\n/U510y7GhGmrQ1fxg+vat642r7jxheJ9N74DaaNMk6KQpgKSO6UAqaYLGxJ99G4Si4G0rWnu\nWuP0fXEhbc05+R1jt7aAZDpHrvYwdl3ksd6luRlfcd6VvW98MYcdEe/3Ht0CeSElJHnsbtYh\n/I0rU4N0Ggvvqvj1oON4bavs1cbpoWxmEpBMW81r2xTOZTdy84rjXCjOd+M7kDdKnnQ1e904\nfRkguVNKTxHKZ4e/H7pS+mlExhExkPhNrOdGzt9pHhcSPzPzzNDNcwmS6Ry+bIbYsb6wKTt7\neS2v/V9R+NbiS+yQlaEbJwdHdn/Lv/dTeMMWyAspIZnG7pcTulDt49nssVQgfczaLOJ8Zwkb\ny/nD7IQtnM/O3rcsul1xamxC3mrOr2JHsq95zIobXyjOd6V/Dzlpo+RJf2MloR/p9azGkCw2\nHClL6Umr89sxlvXbVowN350fA2lrD5Z33CHs2GPDkJpFdpH3uCb69/ssC/8JSpDkc+QHZEO/\n+paMtega+nCe8TyKuotY7jE9ctifuPn3XspyBz5TvwXyQmpI8th7WPMBAzu3GMw6TFBAKo7+\nKD2ejnmK0O0s43entGQnhq6kK89g+Sf0yMh8rX674tTYhLzVxs4BdmRkSWnFjS8U57vSv4ec\ntFHypDWF7NAhvdghsZAsNxwpS+1lFLv/0Xvf3A5D5nH+4PNmSLziz8c2K7qhrG8YUn39on+/\n2/PYg9wEST7HDIlvvuusTs2OOH929MvnTm/dpp/x8Lz5D31cq2Zj6rdAXkgNSR5bO/HI/N/9\naU3ZuQUXKCA1dF/sc+3eLmlf0OOR8I3dugl9W7YrWcjrtytOcUxIWx3amgPZ36OLihXHu1Cj\n70r/HqakjZInfXX2/owduzwWkuWGI2V4Yd9e3La1Xm9B+gRICLkQIKVfZRul4rwEBREESOnX\nfdI9OtO9TkQXICHkQoCEkAsBEkIuBEgIuRAgIeRCgISQCwESQi4ESAi5ECAh5EKAhJALpQCp\nh+nFrHWjCgfyP3ayWcqqb0qTn4+Q/0oBUt//k7+axQZ+EAPp3aG7YpeyqKJAetHQtuGdii7a\nyPnS+qeJ4WUxKDA5vmk3yXjJthnS38LvIZRAlddKr74rO6TgymtbddjC1w8P1y9yyG+EglDq\nkGoj747wOFuXGKTa2HdT2NRvPyZBesQ4duhnmffWf33aKXUcoYCUAqSeoXs//c6ZegBrc/mO\n8LsX/CEC6e0+B7Q46onQjbrQWUPCS/Evz2rd9izjgCYNy3Oe3Sa6ntLrrz9PgnTI4cbH41tH\nv3yyAC/fRMEpVUgHNbnqyQvYcL7kRvbKkjCkZ9kRt9zajb3EF49ib6wILzUnt8Ptt3fInS0t\nz3nREWJVcwSk6szwm4mMDh+WgPP1BY86+8EQ0lmqkIzDvPMeHaSbdqd3MI6jUzAietMutFTt\nkQeWcr65bfc6aXlTEqSf2d3GpwlsWfjLYQdVpfojIaS/VCE1N47zdtl+EqRdxntUrG82RED6\nwTjiDud3s9XS8qYkSIvYP4xPk9kHxqflWVNS+nEQ8qZUIXU1Tg2XIfFFjw4vzmISpJnhd8Ti\nr7E50vKmJEhrWHg3wyNssfFpUHvVOzAj5LdShdTdOGWCdG9Gx+teWd3RBOkNY6npbIa0vCkJ\nUgW7zvh0JzPegXJzDh5DQoHKNUg7c64wzmpnuml3v3HWfezHBCDxg4uNjycfYOz0/iv7gSMU\noFyDtCR8H2deRhjS5sjOhsOLtnK+pejw2kQgjc9YwPmKnNuN08VdU/xxEPIm1yBVddr3tqev\n2K9N5zmh+zl/nhdealbOQWPv7JQ7R16edzxKrCoK6b/tHg6BO7TtA3/t0Ml4T7rSjCuc/2QI\nacy9+0jL+xW2v/Dn/xxwBl/Tt9lVkQdkF/Rr3frMRdwEqeEBWd4A6XE2LvRx85AORYPWGV+/\nyP7j7KdCSHN4GQVCLgRICLkQICHkQoCEkAsBEkIuBEgIuRAgIeRCgISQCwESQi4ESAi5ECAh\n5EKAhJALARJCLgRICLlQ8pB22VZeVW6/kLvt0T+xarf2kfonVpXpHlmhf6KDH3K3A0iltpXx\nnfYLuVu59om7+XbdI/fon8h/1T2ycpv2iXxLypf9FZCcBkg0AZIIkGgCJJqJgCQCJJqJgKQO\nkBwHSDQBkgiQaAIkmomAJAIkmomApA6QHAdINAGSCJBoAiSaiYAkAiSaiYCkDpAcB0g0AZII\nkGgCJJqJgCQCJJqJgKQOkBwHSDQBkgiQaAIkmomAJAIkmomApA6QHAdINAGSCJBoAiSaiYAk\nAiSaiYCkDpAcB0g0AZIIkGgCJJqJgCQCJJqJgKQOkBwHSDQBkgiQaAIkmomAJAIkmomApA6Q\nHAdINAGSCJBoAiSaiYAkAiSaiYCkDpAcB0g0eQNp86OnHHXhZ8lfFpAcB0g0eQPpIhYq792k\nLwtIjgMkmjyBNJ2FOzTpywKS4wCJJk8g3RKBxJYle1lAchwg0eQppOXJXhaQHAdINOGmnQiQ\naAIkmonY2SACJJqJewukTY+e3O2P2P0NSFQT9xZIqQZIjgMkmgBJBEg0ARLNREASARLNREBS\nB0iOAySaAEkESDQBEs1EQBIBEs1EQFIHSI4DJJoASQRINAESzURAEgESzURAUgdIjgMkmgBJ\nBEg0ARLNREASARLNREBSlxykj24eOGY9IJkDJJrSGNLcC2YuvmNUHSCZAiSa0hjSla9zvnHM\nBkAyBUg0pS+kn0o2m762Xzsg0QRINBM1QVrc/+NrBo1Za5x8ZtSoUbdU2VbDa+wXcrda/RN5\ntfaR+ifuDT9kHU/9spVJQJrbf+TnK+69ZHfo5Oji4uJTbS+A0N5SbcMpe0jzS5ZzvueCD0In\ny3fs2LFzi21lfJf9Qu62R/vEcr5D98gK/RP5Nt0jK7drn8i3pnzZbUlA+rbEuDK6clr91/Y3\nHHEfiSbcR6KZqOk+UsWgrzkvG/AZIJkCJJrSFxKfPHzBt3deWQNIpgCJpjSGVPfs8MEPbG34\n0n7tgEQTINFMxHPtRIBEMxGQ1AGS4wCJJkASARJNgEQzEZBEgEQzEZDUAZLjAIkmQBIBEk2A\nRDMRkESARDMRkNQBkuMAiSZAEgESTYBEMxGQRIBEMxGQ1AGS4wCJJkASARJNgEQzEZBEgEQz\nEZDUAZLjAIkmQBIBEk2ARDMRkESARDMRkNQBkuMAiSZAEgESTYBEMxGQRIBEMxGQ1AGS4wCJ\nJkASARJNgEQzEZBEgEQzEZDUAZLjAIkmQBIBEk2ARDMRkESARDMRkNQBkuMAiSZAEgESTYBE\nMxGQRIBEMxGQ1AGS4wCJJkASARJNgEQzEZBEgEQzEZDUAZLjAIkmQBIBEk2ARDMRkESARDMR\nkNQBkuMAiSZAEgESTYBEMxGQRIBEMxGQ1AGS4wCJJkASARJNgEQzEZBEgEQzEZDUAZLjAIkm\nQBIBEk2ARDMRkESARDMRkNQBkuMAiSZAEgESTYBEMxGQRIBEMxGQ1AGS4wCJJkASARJNgEQz\nEZBEgEQzMXiQvr7qlD++nNxEQBIBEs3EwEF6P5+FuiGpiYAkAiSaiYGD1IWFm5PMREASARLN\nxKBBWhpxxEYnMxGQRIBEMzFokL6KQrotmYmAJAIkmolBg7SpKAJpWjITAUkESDQTgwapdGrY\n0blJTQQkESDRTPQtpMn/13v4V/G+8fapRd3v/SWpiYAkAiSaiX6FNMK43mn6vjsTAUkESDQT\nfQrp7cg9oa7uTAQkESDRTPQppNuiO+dWuDIRkESARDPRp5BuikJa6spEQBIBEs1En0J6OeLo\nIHcmApIIkGgm+hRS6blhSNPdmQhIIkCimehXSBvGH3dwyXsuTQQkESDRTPQrJFcnApIIkGgm\nApI6QHIcINEESCJAogmQaCYCkgiQaCYCkjpAchwg0QRIIkCiCZBoJgKSCJBoJgKSOkByHCDR\nBEgiQKIJkGgmApIIkGgmApI6QHIcINEESCJAogmQaCYCkgiQaCYCkjpAchwg0QRIIkCiCZBo\nJgKSCJBoJgKSOkByHCDRZAHpi6nzNlNNBCQRINFM9Amk7/sxxo5ZSDQRkESARDPRJ5DODh/r\n5MikjkSc+ERAEgESzURPIX21YFPkxMLoYeySe0vLhCcCkgiQaCZ6CGl6Z8YOeCJyMgrpYZqJ\ngCQCJJqJ3kGaH34zWPaqcfqLKKSXaCYCkgiQaCbqh/T2tcMf3xj6/KeInV7hM08Nnz5sPclE\nQJICJJqJ2iENNsT0+Lm09IQIpNbhc1ecaOxr+IxmJCBJARLNRN2QnojwubK09BzzO7fMnTxr\nE9FMQJICJJqJuiH1j/Dp0HCc/PH0MwFJCpBoJuqGFLkzxPYNnRydGzoxhOrpDFKAJAVINBN1\nQ7o+AqmvcXrhhIc+1DETkKQAiWaibkjfFxmOmnygcyYgSQESzURtkD7oV3TY9WtCV0Pntsjt\n9Y6moZEASQqQaCbqgjQ7z7gmOm6D8YDsBj0jGwIkKUCimagLUnHkvtEEvB5JDpBoSl9IGzMj\nkC4CJDlAoil9IW3KjUC6bG+CtN22cr7bfiF3q9A/kZfpHlmpfyLfpWdQvwikqdu3V2uaKKrm\nO1K+7E4HkPbYVsWr7Bdyt2r9E3ml7pE1+ifyCj2DfmhtOPpT6FStpomiWic/pANI9td3uGlH\nU/retCstXXlbv0GTjRN7z007+7UDEk3pDKkhQBIBEk2ARDMRkESARDNRA6RFL70vPwQLSCJA\noikdIa0bxBj73XviDEASARJN6QhpRHi3d7sfGs4AJBEg0ZRmkN698IRLZkQfif1Hw7mAJAIk\nmtIL0gQDUNQRu7HhbEASARJNaQVpeVMm9feG8wFJBEg0pRWkybKjtt83nA9IIkCiKa0gPRU1\n1Cv030EzxPmAJAIkmtIK0qKIo7yVnz/7tnzgR0ASARJNaQRp1drS28OQHoj9DiCJAImmtIH0\nymEs85g5T/Zu3/f5Rt8DJBEg0ZQWkH75+POZ4QM0tPwq7vcBSQRINAUd0jfjRoz/x/6MNYnc\nPRoadyFAEgESTQGH9GI+M9Uz7lKAJAIkmoINaWUrsyN2ZtzFAEkESDQFG9LkGEfsqbiLAZII\nkGgKMKT1Y7vtE+Po8vhLApIIkGgKKqSNbz/e24xo9B1jZlksDEgiQKIpiJA2PnbZ4I4x10W5\noxUXACQRINEUQEjrjjEbajPu00cnLlJdApBEgERT8CD9dJHZ0WL7iwCSCJBoChak/51z9G+y\nzI66J3AxQBIBEk2BgvRw7N5uxlp8lMDlAEkESDQFBtLyV9+6PztG0YGnXrs0kcsCkgiQaAoI\npM3X5DS+NjrzuwQvDUgiQKIpIJDujTHUtOSN15Q76kwBkgiQaAoIpLZmR21XJXVpQBIBEk3B\ngLQxxtHbyV0ckESARFMwIC0paEDUc8z9k9ckeXFAEgESTYGA9GgzsYNhcwojAUkESDQFAdKb\n0T12GSz30uTuHEUDJBEg0eR7SJvmnh19e/Lma5ZssF8+XoAkAiSafA1pw+cPny9eA5vxS6oj\nAUkESDT5GdLE/Uz76jqmPBKQRIBEk38h/XRTzIOwj6Y8EpBEgESTXyFNPSyGUd79qY8EJBEg\n0eRTSP+LfV5dYQIvO7IMkESARJNPIRXHOGr7mpORgCQCJJp8Csn0jmHs8OeSfS6DOUASARJN\nPoV0gLhvdNTYRF67pwyQRIBEk08hnRNl1GquGyMBSQRINPkR0opLokf0zjxtvisjAUkESDT5\nD9Kax1pGr45uS/mpDDEBkgiQaPIbpHXDMxvuHvVzayQgiQCJJr9BGi7trDvWrZGAJAIkmnwG\n6btMCdIQt0YCkgiQaPIVpPWjW0iOmn/h1khAEgESTb6CdK78KGyXJA/MoAiQRIBEk38gbZg3\nRmJ09jcujgQkESDR5AdIS8cUt27ZNENi1OkJV0cCkgiQaPIBpCnNWEznuTwSkESARJP3kL7I\ni3XU9GOXRwKSCJBo8hzSzwfGOiqa5vZIQBIBEk2eQxod6+gS90cCkgiQaPIc0kkxjlo6eSms\nRYAkAiSavIa0NOYeUofpBCMBSQRINHkNqf4KKTOvxaEXPTBpznqKkYAkAiSaPIb0atRRs0Tf\nMyylAEkESDR5C2lTu/onMpCOBCQRINHkLaT59XeNxpOOBCQRINHkLaT6Vx+1d+u1sPEDJBEg\n0eQppL/XXyG590TvuAGSCJBo8hDSoiH1r+L7A/FIQBIBEk0eQfr6rCbSg0evEo8EJBEg0eQN\npK8KJEas2Y/EIwFJBEg0eQJpRVfTsxn+ST0SkESARJMXkO7MNDkaRj4SkESARJMHkM6LeXZd\nKu9TnlyAJAIkmrRD2ny82VGeK0f3VgdIIkCiSTsk05GCWO4pbr8aNl6AJAIkmjRD+uogSVG7\n8aRPVRUBkgiQaNILaV6u5KjlEl1jAUkESDRphbT5AMlRl/e1zQUkESDRpBXSvYLR2dQPwsoB\nkgiQaNIIafO/sxocjdQ21QiQRIBEkz5I644Q10cn6BoaCZBEgESTPkhnCEdnrtM1NBIgiQCJ\nJm2Q5gtHT2ka2RAgiQCJJl2QvuvU4KhrAu9q7m6AJAIkmvRA2lgi7fd+CpCUAZLj0hXSypaS\no9PqAEkZIDkuTSFt7iwY5f+tXPlmzCQBkgiQaNIBabJ0fTTL5l3NSQIkESDRpANSkXB0us27\nmtMESCJAokkDJGlHwzGbAMk2QHJcWkL6VwOjnPDxVAHJJkByXDpC2rhPvaN+GyITAUkdIDku\nHSHNqXfUZGV0IiCpAyTHpR+kxYc03LAbVz8RkNQBkuPSDtLDGQ2OTmqYCEjqAMlx6QZpmthf\nVyAmApI6QHJcekFa2EN6IPYsMRGQ1AGS49IK0qVM7iUxEZDUAZLj0glSf5OjvtJEQFIHSI5L\nH0gvNzE5GrBJmghI6gDJcWkD6WxZUcZVpiMGAZJNyUJaM2IXIJlLD0gbSjJMV0ePx0wEJHVJ\nQqq6pmQnIJlLC0jTzG/awgbFTgQkdUlCemoUIMWWDpA+N10dZbT/b6OJgKQuOUiLh34JSLGl\nAaRfcmRHBRvjTAQkdUlB2jn0y5URSM+MGjXqlirbaniN/ULuVqt/Iq/WPtLdiRWmN4ftvC3e\nxMD/kAlUx1O/bGUykMY/zqOQRhcXF5+awFUYCkIdJEY5H3m9NcGstuGUPaT3RlXWQyrfsWPH\nzi22lfFd9gu52x7tE8v5Dt0jK9yc+JB8/+jQDRYT+TYXRyZU5XbtE/nWlC+7LQlIE0vCTaj/\n2v6GI+4j0eTmfaR+EqN9n7eciPtI6pK5j7R17dq1H5Us2wJIpoINaaLkaJz1YoBkU7IPyK7E\nXrvYgg2ptXDUTzURkNQBkuMCDWmBcHSwciIgqcNz7RwXaEhix3cr5bu2AJJNgOS4IEN6sMFR\n31/UEwFJHSA5LsCQ1jY8w+5du4mApA6QHBdgSOfXO2pjOxGQ1AGS44ILaULDDbv5thMBSR0g\nOS6wkNo1OOphPxGQ1AGS4wIK6THxzKCMz+0nApI6QHJcICFtPEx6RsN5CUwEJHWA5LhAQjpf\ncpT5XQITAUkdIDkuiJDWya8svz+RiYCkzgpS1bzxVw8cMfZ/OwDJriBCkp4ZxMYkNBGQ1MWH\ntOTy5qxp+yN/U8iyzpxRB0jKggjpyQZGud8kNhGQ1MWDtGlw9plPLg37+fmVy1r0+AqQVAUQ\nUt+EnvFtmghI6uJBanf7RolL+aROiwBJUfAgDWxwNCPhiYCkLh6kjTFg9mwHJEWBg7RM3D/6\nIuGJgKQOe+0cFzRIj4g9dtkbEp4ISOriQeppCpBsChakXtL+OnZN4hMBSV08SL1NAZJNQYI0\nXGbETkliIiCpw007xwUIUpHJ0cxkJgKSOjtIc84AJJuCA+kIk6PhSU0EJHWWkF4aMcToNwcA\nkk2BgTTd5Cg34R0N4YmApM4K0iRW0Ix1bJfR5r+AZFNgIGXLjrJ/Sm4iIKmzgtStW+W6zDV8\n3n4/A5JNAYG0qbns6OTNSU4EJHVWkJrfxvnR0zi/fDAg2RQQSAcLRflLkp8ISOqsIO1/C+fD\nruf86Q6AZFMwIE0SjlamMhGQ1FlBOvHISv7Pgzi/tSUg2RQISHcJR6+lNBGQ1FlBept1KFuU\nccU/9j0TkGwKAqS/CketU5sISOosd38/e/oufk8OO2gFINnkf0g/NxGOMtekNhGQ1KkfkN25\ntMraESBF8j2kp6S9dRnLU5wISOqsIJ3xQrmCECBJ+RzSOtMbxH6Q6kRAUmcFKZ8VDJtr+SJz\nQJLyN6TV8htbsptSnghI6qwg7X5lYD7rNHYlINnmZ0ir9jE9LegvqU8EJHWK+0jl0wbls+Mn\nAZJNPoZkvjpiJzqYCEjq1DsbdozMULy6wn7tgERTgpA2mx113ehgIiCpU0Da/frF+7CWQwHJ\nJt9C+iVLZtTyR0cTAUmdFaRf/3NuM1Zw8ZuV1o4AKZJfIW02OVK8Y3lCEwFJnRWkbNZ88PQK\nhSJAqs+vkAolRnmOJwKSOitIg17F40gJ5k9ID8hXR/bvf2Q7EZDUWd9H2jzl7tHPrQck23wJ\nqZPEKMvRvaPoREBSZwlpfJ4wZlAAACAASURBVL7xO2h6DyDZ5UdIx0uOMje5MRGQ1FlBeo79\n/t2NpbNPYJMBySYfQtrPhWfXxUwEJHVWkI47fI/xqeJIHCDSLv9BypMduTQRkNRZQKrLGxM5\ncVcLQLLJd5CkF00wttCliYCkzgJSVc4VkROjegCSTX6D1FG+PnrBrYmApM7qpt2QJu8Znz5o\nOhmQbPIZpAclR+e6NxGQ1FlB+ud+rM81V/dhRWNCvQlIivwF6SfJUepP9m48EZDUWUEyPdvx\nakBS5C9I0lG3Jrk5EZDUWUGqkasFJEW+gjRHOHLj4SMxEZDUxYMU+3SG8m2ApMhPkDYKR+vc\nnQhI6uJBan/bBolL2cQOeA9ZVX6C1Iri/lF4IiCpiwdp85DsfpOWhG/PrZ16SfNjv47vCJAi\n+QjSrQ2Omro9EZDUxb+PtGxkAcsr6toxn2WfPdOCESBF8w8kcX3EXnR7IiCps9rZUP3Jg9de\nOHLcW7ssGQFSNL9AWie9sLyl6xMBSR3e+tJxfoGUIxxlOzg6g8VEQFIHSI7zCST5bcQIJgKS\nOitI3aMdf/7NawFJmT8gyQcMeohgIiCps4I0vA0r7H70PuzgI/PzZgOSKl9Akm7Xsd4UEwFJ\nnRWkl7MnVXFe/XS7ZTv/2M7i0MX2awckmhpBOk5ydCnJREBSZ/nCvuGRz5efzn9kqwBJkQ8g\nLZMcLaCZCEjqrCAV3hv5/JeWfAd7H5AU+QCSdAdpMdFEQFJnBemko8KH49pT/Hs+B9dIyryH\ntL9w9D7VREBSZwXpo9wuE95955HDs+d+3L53fEeAFMlzSDcLRy4ceMtiIiCps3wc6b1i4xdz\n2Bw+qe86QFLlNaT1wtF8uomApE7xgOyqmW+uqOHc+t3G7NcOSDSZIIk7SMcQTgQkddaQ6lbP\nmfGj8j377NcOSDRJkOaL6yOXjrwVfyIgqbOENLub8bs54j1AsstTSD9KO77XU04EJHVWkBbm\nHnj3a9PvOzD3K0CyyVNI0o7vp0gnApI6K0j9OpYan7Z0PBuQbPIS0m3CURvaiYCkzgpS6zsi\nn0e3ASSbPIS0Vs8dpFJAss0S0p+jkFoDkk3eQdqkzREg2WV9026L8WlLpzMBySbPIJ0vOXL1\n2FvxJgKSOuudDUX3vzH9/qLchYBkk1eQCqQddvQTAUmd9e7vI4xf0OGzrB0BUiSPIDWXHN1F\nPxGQ1Fk/IFv7w8yZP1gcYxWQpLyBdIHkqJmGiYCkDsdscJwnkJ6VHGXqmAhI6uJB6mkKkGzy\nApL8XmLUO+wiEwFJXTxIvU0Bkk36IUlHymcsV8tIQLIJN+0cpx3SO7Kje/TMBCSbVJAW/tro\nLEBqnG5ID8uOntE0FJBsUkFi0wApgTRDOl129BtdUwHJJkBynF5IebKjFtrGApJNgOQ4nZBW\nywdUZa21zQUkuwDJcRohtZUZZczWNbYUkGxTQVq6A5ASSB+k02RHWZqGRgIkm7D723HaIF2j\n/+GjhgDJJkBynC5IPWVHB2gZKQIkmwDJcZogZcqOrtQxUQ6QbAIkx+mBZNpd963+Z/cBkjpA\ncpwOSPNlRtmW72pOFyDZFA/SdlOAZJMGSL/Ijppavqs5YYBkUzxIzBQg2UQP6RP595FTCkhU\nE12G9LdQDx3ETrl53AXZvd8CJJvoIcmO9jfOACSaiQT3kf7VbK7xaXGLfwKSTdSQ7pIdnRA+\nC5BoJhJAOnpk5PM1RwOSTcSQ7pYdbYicB0g0EwkgtRgd+XxnASDZRAtJPuhWRtQRIBFNJIDU\n67Ay49Puw/BSc7tIIckPH4mDMwASzUQCSFPZ0a+tXv1aMXsJkGwihLQgviNAIppI8YDshPBt\nipaPWDsCpEh0kK4wPQ4hfQOQaCaSPLNhy7TxD722TeGI77Stgu+xX8jdKvVP5LtpVnyQzChD\n/k5VOc1E66p4me6R1fon8l0pX3aXJaSds17YUKZ868s9tlXxKvuF3K1a/0ReSbLefNPTgkzf\nqqGZqKiGV+geWat/opMf0grS46Hf44dT2r+sgGR/fYebdqmXLTsqNH8PN+1oJhLctHuT9X2R\nffjtcRkzAMkmGkhFsqPYd7UEJJqJBJB6d6/h7ENeeXQfQLKJBNJi+e7RmtjvAhLNRIoHZO/m\nBiT+YEtAsokCkvxeE3GOkQ9INBMJIHW4IwLpuvaAZBMBpBskR/HeswWQaCYSQBrYbpsBaVXr\n8wHJJvchvSA56hpvAUCimUgAaXVBp/vZmLvatFgJSDa5DqmN5GhQ3CUAiWYixQOy3/Q1fo+n\nLbZ2BEiR3IYkH+XE4qBbgEQzkeaYDb9+9qXideaAVJ/LkCyeXmcKkGgmEkA644VyJSJAashd\nSIk4AiSiiQSQ8lnBsLnKJwgBUjRXISXkCJCIJhJA2v3KwHzWaaxiVwMgRXMTEkvIESARTaS5\nj1Q+bVA+O34SINnkIiT5+kj1XuWARDOR7ACRO0Zm4HBcdrkHSX6eal/VgoBEM5EG0u7XL96H\ntRwKSDa5Bkne732ncklAoplIAOnX/5zbjBVc/GaltSNAiuQSpLXy7br/qZcFJJqJBJCyWfPB\n0ysUigCpPncgvS/vZ3jNZmFAoplIAGnQq3gcKcHcgSQ7sl0YkGgm4t0oRAGFJN+uu9x2aUCi\nmeg+pG2dnwKkBHMB0gb5+qin/fKARDOR4BrpdMXeOkAy5QIk2dHFCSwPSDQTCSB93/XJGkBK\nKMeQPpMd3ZDIJQCJZiIBpAGnseZdehgBkk1OIZ2d2NOC5ACJZiIBpL4NAZJNDiH9Jqn9dZEA\niWYi9tqJggZpf9nRzwleCJBoJpJAsj/Sqv3aAcmuJrKjjYleCpBoJlJAwpFWE80JpMReftQo\nQKKZiCOtigIFKUVHgEQ0EUdaFQUJUqqOAIloIo60KgoOpBmJvoyvcYBEMxFHWhUFBlIeS9kR\nIBFNxJFWRUGBJL+Kz+rwdZYBEs1EHGlVFBBInWRHbZO9NCDRTMSRVkXBgPSl7OjtpC8OSDQT\ncaRVUSAgvSc7mpn8SECimUj1FKGq6W+VAZJdyUPKkh0tTGEkINFMJIBUdtnhvO5kxn73CyDZ\nlDQkeT9DkrvrogESzUQCSDexi/ksdsNrhVcBkk3JQpIfPmqS2khAoplIAKlTf86vaVXBLz4Y\nkGxKEpLz6yNAoppIAKnJvZx3O5/zh5oAkk3JQUr5aUFygEQzkQBS5wv4CjaR8xEdAMmmZCD9\n5IojQCKaSADp1pzrj8hdv+Pu3IsByaYkII128LQgOUCimUgAaWf/jKxH+EJ26GpAsilxSH+W\nHXVzMBKQaCaSPI60vYzzLR/ssXYESJEShnS67OhGJyMBiWYiCaS1k0ff8+KvXJH92gFJqrPs\nqIujkYBEM5EC0h25xu+78J+AZFeCkNrKjl5wNhKQaCYSQHqM9Xp3Y+ms49k0QLIpMUjy0YIy\nfnE4EpBoJhJA6tE1/G4Uew4/AZBsSghSjiu7vesDJJqJFC81HxP5fFcBINmUCKRsVx0BEtFE\nAkjHjYp8vqobINmUAKRMdx0BEtFEAkjP5881Ps1rdhcg2WQPyW1HgEQ00WVI44y6Zpxyw/Wn\nZDR/GpBssoWU+tGCrAIkmokuQ2KmAMkmO0juOwIkookuQ6oxBUg2qSH9Xf6fUpZLIwGJZiLB\nfaQlL99z498XWSMCpIaUkJ6RHWW7NRKQaCa6Dmlut8hvvvtcQLJNBWmF7KiVayMBiWai25Ce\nz8oaNnXRNy9fkZ39AiDZpYIkOzrAvZGARDPRZUg/5rVfHjm1vGPBZkCySQFJ3s/g7Gmq5gCJ\nZqLLkK7K+qT+5CfZNwCSTdaQ5Ouj8W6OBCSaiS5D+u1xAkvP47hl9mvfuyFJ10euPAwrAiSa\niS5DajJSYLm6EJBssoJE5wiQiCa6DKlgiMByCd4fya74kNYTOgIkookuQ+pxiMDyu2MByaa4\nkL6ldARIRBNdhvQA+3f9ycfYA4BkUzxI30q7GTI2uD4SkGgmugyprFPeo9XGiep/NemgOIq+\n/dr3Vkjy84Lcvz4CJKqJbj8gu7w96zzswQeHHczaLbV2BEiRGkOSXw7LKEYCEs1E158itPWm\nNsbfQOsbtigcAVKkRpAyqR0BEtFEiqMI/frxx1tVigCpvlhIppfx/UQyEpBoJlK90RggJVIM\nJHeO7q0OkGgmApLIa0g6HAES0URAEnkMSYsjQCKaCEgiTyHJDx+59bLyeAESzURAEnkJ6WbZ\nUR7hSECimQhIIg8h9ZEdufdy2DgBEs1EQBJ5B0k+mio7mHQkINFMBCSRZ5BMD8O+QjsSkGgm\nApLIK0jy7jr2HPFIQKKZCEgijyDp2e1dHyDRTAQkkTeQ9DoCJKKJgCTyBJJmR4BENBGQRF5A\nIji6tzpAopkISCIPIMmOcrWMBCSaiYAk0g5pk+zoFD0zAYlmIiCJtEOSHU3VNBOQaCYCkkgz\npKbyw0eH6ZoKSDQTAUmkF5LpaUFHaRsLSDQTAUmkFZLJ0bX65gISzURAEumEZHL0hb65gEQ0\nEZBEGiGZjnKyXtvYUkCimghIIn2QTI50DY0ESDQTAUmkDZKHjgCJaCIgiXRBypIcufVm5QkH\nSDQTAUmkCZJ8fZSl/88akEgmApJIDyTT9ZHizZiJAiSaiYAk0gJJvj7KUb2rOVGARDMRkEQ6\nIMnXR3mqdzWnCpBoJgKSSAMk2VGu6l3NyQIkmomAJKKHlGl2BEhEAZIoHSHJL5toapwBSDSl\nMaTKxy4fNObHvRtS45fDAhJNaQzpnks//3bcJbv2Zkim/XWRswCJpvSFVFqygPM9Az7YiyHJ\n10fZ0fMAiab0hbTqhp2c1w15fe+FFM8RIBGVvpDCzS/5PvTxmVGjRt1SZVsNr7FfyN1qCSfK\njnLERF5NNzJ+tfon7g0/ZB1P/bKVyUGqm3neJOPz6OLi4lMTk5c+yY4KvN4Y5K9qG04lAmnj\nbYPeCZ8o37Fjx84ttpXxXfYLudsesomyo47S+eV8B9VIqyr0T+TbdI+s3K59It+a8mW3JQPp\nu4H3bZe+tL/hmEb3kV6RGEUeP6oP95FoSt/7SDVDJ9bJX9uvPX0gTZMdnWD6FiDRlL6QFvT/\nZEmorXsjJNnRAPO3AImm9IX0ekm4t/ZCSNL9o4ynY74HSDSlL6TY7NeeLpBkR42+CUg0AZIo\nPSAtVToCJKIASZQWkF5gSkeARBQgidIB0lwbR4BEFCCJ0gDScjtHgEQUIImCD+llW0eARBQg\niQIPaay9I0AiCpBEgYeUgCNAIgqQRAGHtEm93zsaINEESKKAQ0rIESARBUiiYENKzBEgEQVI\nokBDStARIBEFSKIgQ0rUESARBUiiAENK2BEgEQVIouBCStwRIBEFSKLAQkrCESARBUiioEKS\nj3NiuzAg0QRIooBCkq+PNtsuDUg0AZIomJDk66NN9osDEk2AJAokJNnRUwksD0g0AZIoiJDk\n95sYm8gFAIkmQBIFEJLsaHxClwAkmgBJFDxIkiPb/d7RAIkmQBIFDpJ8fZTAfoZwgEQTIImC\nBkl+v/I+iV4IkGgCJFHAIMmOChO+FCDRBEiiYEGSHTW1X7w+QKIJkESBgiTfP8pL4nKARBMg\niYIEKVVHgEQUIIkCBCllR4BEFCCJggPJ9D7LyV0UkGgCJFFgIGWm7giQiAIkUVAgyY5yk70w\nINEESKKAQEr9/pERINEESKJgQHLmCJCIAiRRICDJj8M2S2EiINEESKIgQJKvjxJ/XpAUINEE\nSKIAQJId5ac0EZBoAiSR/yHJt+uapDYRkGgCJJHvIeU4dwRIRAGSyO+QslN7vrc5QKIJkEQ+\nhyQ7Svb5DCJAogmQRP6GJN+uS+Hxo/oAiSZAEvkakkuOAIkoQBL5GVKOK7frSgGJKkAS+RhS\nnuQoy9FEQKIJkET+heSeI0AiCpBEvoWU654jQCIKkER+hdTERUeARBQgiXwKycXbdaWARBUg\nifwJKd9VR4BEFCCJfAlpH8lRpgsTAYkmQBL5EVJblx0BElGAJPIhpC5uOwIkogBJ5D9It7ju\nCJCIAiSR7yCNdd8RIBEFSCK/QfovgSNAIgqQRD6DtEw6MHGib2xpHyDRBEgif0GaxCgcARJR\ngCTyFaSxNI4AiShAEvkJUlsiR4BEFCCJfATpBCpHgEQUIIn8A6kPmSNAIgqQRL6BtD+dI0Ai\nCpBEfoHUitARIBEFSCKfQCqkdARIRAGSyB+QKG/XlQISVYAk8gUksv3e0QCJJkAS+QHSocSO\nAIkoQBL5ABLhfu9ogEQTIIm8h9SF3BEgEQVIIs8hUd8/MgIkmgBJ5DWkjhocARJRgCTyGNLx\nOhwBElGAJPIW0rlaHAESUYAk8hQS/f66SIBEEyCJvIT0liZHgEQUIIk8hLSB5PgM8QIkmgBJ\n5B2kL7U5AiSiAEnkGaRV+hwBElGAJPIMkkZHgEQUIIm8gqTTESARBUgijyBpdQRIRAGSyBtI\neh0BElGAJPIEkmZHgEQUIIm8gKTbESARBUgiDyBpdwRIRAGSSDukr/U7AiSiAEmkG9IiDxwB\nElGAJNINyQtHgEQUIIk0Q/LEESARBUgirZDGeuMIkIgCJJFOSFTvI2YbINEESCKNkDxzBEhE\nAZJIH6TTPXMESEQBkkgbJO+ujwCJKkAS6YJ0l0f7GcIBEk2AJNIEqb98faT9SUmARNPeA2mn\nbRV8j/1CjjMdd6tSx0RTlXy37pFV5don8jLdI6v1T+S7Ur7sLgeQ9thWxavsF3Kc7GhPtY6J\npqp5pe6RNfon8grdI2v1T3TyQzqAZH99p+Wmnfn+kcW7mhOGm3Y07T037ezXrgHSQzH7GQCJ\nZiIgqQs8pNj9dYBEMxGQ1AUdUqP93oBEMxGQ1AUcUuPHjwCJZiIgqQs2pDiPwwISzURAUhdo\nSPGezwBINBMBSV2QIcV9XhAg0UwEJHUBhhT/+XWARDMRkNQFF5LF81QBiWYiIKkLLCSr53sD\nEs1EQFIXVEiWr5sAJJqJgKQumJBmWb/+CJBoJgKSukBCUh0vCJBoJgKSuiBCUr6uHJBoJgKS\nugBCUh+fAZBoJgKSuuBBsjnOCSDRTAQkdYGDZHc8VUCimQhI6gIHye54QYBEMxGQ1AUNku1x\ntwCJZiIgqQsYJPvj1wESzURAUhcsSAkcBxKQaCYCkrpAQUrkeKqARDMRkNQFCVJCxyUGJJqJ\ngKQuQJASO743INFMBCR1wYGU4HHyAYlmIiCpCwqkFxN9vwlAopkISOoCAinx94cFJJqJgKQu\nIJASf/8jQKKZCEjqggEpifcRAySaiYCkLhCQknk/PkCimQhI6oIAKan3tQQkmomApC4AkJJ7\nf1hAopkISOr8DynJ91kGJJqJgKTO95CSfb9yQKKZCEjq/A4pWUeARDQRkNT5HFLSjgCJaCIg\nqfM3pOQdARLRREBS52tIKTgCJKKJgKTOz5BScQRIRBMBSZ2PIaXkCJCIJgKSOv9CSs0RIBFN\nBCR1voWUoiNAIpoISOr8Ckl29LdkLghINBMBSZ1PIUmO2NikLglINBMBSZ0/IcnXR8k5AiSi\niYCkzpeQZEdvJHlZQKKZCEjq/Agp1f0M4QCJZiIgqfMhJAe360oBiWoiIKnzHyRnjgCJaCIg\nqfMdpNT310UCJJqJgKTOb5CcOgIkoomApM5nkBzerisFJKqJgKTOV5ASP56qdYBEMxGQ1PkJ\nks37lScWINFMBCR1PoLUxw1HgEQ0EZDU+QeSO44AiWgiIKnzDaQu7jgCJKKJgKTOL5DccgRI\nRBMBSZ1PIHV0yxEgEU0EJHX+gNTWNUeARDQRkNT5AlKhe44AiWgiIKnzA6TmLjoCJKKJgKTO\nB5Dy3XQESEQTAUmd95CauuoIkIgmApI6zyE1cdcRIBFNBCR1XkPKc9kRIBFNBCR1HkPKddsR\nIBFNBCR13kLKkRxlpvwzmAIkmomApM5TSASOAIloIiCp8xJSNoEjQCKaCEjqPIRE4giQiCYC\nkjrvIGVJjrJS3v5GARLNREBS5xmkTBpHgEQ0EZDUeQWJyhEgEU0EJHUeQSJzBEhEEwFJnTeQ\nZEfZKW973ACJZiIgqfMEUgadI0AimghI6ryAJDvKSXnLLQIkmomApM4DSLKjvJQ33CpAopkI\nSOr0Q5IdNU15uy0DJJqJgKROOyTZUX7Km20dINFMBCR1uiHJjgpS3mpFgEQzEZDUaYYkO2qV\n8karAiSaiYCkTisk+W1bWNuUt1kZINFMBCR1OiHJh8lnHVPeZHWARDMRkNRphCQf3pv1SXmL\nbQIkmomApE4fJNNhiVN7W8tEAiSaiYCkThukQj2OAIloIiCp0wXJ3cOpKgIkmomApE4TJLcP\nA2kdINFMBCR1eiC5f/g6ywCJZiIgqdMCKUefI0AimghI6nRAytboCJCIJgKSOg2QsnQ6AiSi\niYCkjh6S7Mi9w9dZBkg0EwFJHTmkTL2OAIloIiCpo4ZEd7ggiwCJZiIgqSOGlKHbESARTQQk\ndUlBqpsy7NKna5KARHm4IIsAiWYiIKlLCtLUwZ9+ccnTiUMiPVyQRYBEMxGQ1CUDqeaSGZx/\nOGhPopBkR7kpb2GSARLNREBSlwykNSWlnO8qWZYgJPl5QU1S3sBkAySaiYCkLhlIi/rXhj4O\n+Dj04ZlRo0bdUqVOukLKt1nUxWpr9M2KTuTV2kfqn7g3/JB1PPXLViYBae4FxseL3wl9GF1c\nXHyqzeICUqHtqhEKdrUNp+whLexfF/o44KP6r22u7Bqe0rB/yteXKYSbdjQTcdNOXTI37VaV\nhJYuL1mSIKTSqKMuKW9cKgESzURAUpfUXrsh73H+6cCKRCF1NG7cUb6sPF6ARDMRkNQl9TjS\nC5d+t3LEUw1f2q9d/a7mJAESzURAUpfcMxueG3bpU+Jelf3aAYkmQKKZ6N/n2gESSYBEMxGQ\nRIBEMxGQ1AGS4wCJJkASARJNgEQzEZBEgEQzEZDUAZLjAIkmQBIBEk2ARDMRkESARDMRkNQB\nkuMAiSZAEgESTYBEMxGQRIBEMxGQ1AGS4wCJJkASARJNgEQzEZBEgEQzEZDUAZLjAIkmQBIB\nEk2ARDMRkESARDMRkNQBkuMAiSZAEgESTYBEMxGQRIBEMxGQ1AGS4wCJJkASARJNgEQzEZBE\ngEQzEZDUAZLjAIkmQBIBEk2ARDMRkESARDMRkNQ5gbTdttWf/mS/kLuVl+me+P2nv+geuWeX\n7onLP92se2TFTt0Tl3y6NeXL7nQAyb4Xi98lWKvP+nfxF15vAn33FK/yehPou654uwtrAaQU\nA6R0CZA8DZDSJUDyNEBKl/wLae3sDQRr9Vk/zN7q9SbQt2x2mdebQN9Xs6tcWAsFJIT2ugAJ\nIRcCJIRcyH1IdVOGXfp0jeur9VeVj10+aMyPXm8FfWtG7PJ6E6j76OaBY9Y7X437kKYO/vSL\nS552fbX+6p5LP/923CVp/0dWdU3JTvulAt3cC2YuvmNUneP1uA6p5pIZnH84aI/b6/VVpSUL\nON8z4AOvt4O6p0alPaQrX+d84xjnu5ldh7SmpJTzXSXL3F6vr1p1Q+jvq27I615vB3GLh36Z\n7pB+Ktnszopch7Sof23o44CP3V6v/5pf8r3Xm0DbzqFfrkx3SIv7f3zNoDFrna/IdUhzLzA+\nXvyO2+v1W3Uzz5vk9TYQN/5xnvaQ5vYf+fmKey/Z7XhFrkNa2N+44zbgI7fX67M23jYo3f9f\n8d6oyvSHNL9keeje7gUfOF6R65BWlfzKeXnJErfX66++G3ifG0/Q8nUTS8JN8Ho7SPu2xLgy\nunKa4xW5v9duyHucfzqwwu31+qqaoROd7zD1e1vXrl37UcmyLV5vB2kVg77mvGzAZ45X5P7j\nSC9c+t3KEU+5vlpftaD/J0tCpf/TVtP+ph2fPHzBt3de6fwJBATPbHhu2KVP1bq+Wl/1euRW\nz1tebwd56Q+p7tnhgx9w4f+IeK4dQi4ESAi5ECAh5EKAhJALARJCLgRICLkQICHkQoCEkAsB\nkr6msGjtHK3mb8z0NL/ePa0WLG3/i3pNoYtuaLfJ0cagaICkryns3DHh/proJd4dGufV7AlD\nGnKrzeqNi942ONGNQaoASV9T2NQkL/E3Fucpo4lCWpZtc4UUvuiG7KVJbhSKFyDpywpSteUz\nyR1BGtnPbvXhi555udW3URIBkr4kSAuybgp9vD9zXg3758U5TXo9Z5y5+o+dCk4MPxH2szP2\n/e1lW3jf0P2pIfL5L/UuKP53PEhfntW67VlfGl9+dHJhz5eHdw/5KZjEuVj9wBzjTbF254d0\nvd3ngBZHPVF/0Sebu3HE3r0+QNKXfI10a9aX/Psm14f+0vfPGzG6KxvP+TeFRbff3S0j9Of/\nVk7Xu24s6PTr4lHsjRXS+Y+wQ/98RdODGkOak9vh9ts75M7m/P287vdclt0mBOkTtoRLq3+V\n/Se08CvsBf4sO+KWW7uxl6KQlrK94Pga9AGSvhr22g3lfM8hR1f3PXh36C+dvcd5ea/mm/lJ\nHbeFrkf65O+sPuSIMs5nswmRm3YN528tPHoX53NZI0i1Rx5Yyvnmtt3reI8u5ZxPYiFIf82r\n5dLqy5ufG1p4YEE5P71DJeeVBSOikOqajvfmnyO9AiR9Ney1ezX0xUcZfTLnGX/pJxrfeodN\n+ZXdZ5x6kc2az8Kvi5z4YRiSOH8aMy7IT24E6YfIInez1auY8crwqoIQpGuLuLx6PrhZOS/P\nH8b5LuMdJtY3G1J/96r91Tp++HQPkPRl3tlwJbuKG3/p4T/jDezuz+qvr154ntW/8tmAJM4f\nz8IHSb45CmmpceZCQ8NMNt044zU2ZyYLH5KlWwjShV1Nq+fT2et8Gpsb+mrRo8OLs1gDpCMH\n0v/o6R8g6csM6Wx2fJ3xl36N8UUpG/0lG/NhuA2T2YLoMgYkcf5DbLVx5rgopB3TQv0agfSG\nccZ0NuN/LPweb0eHRBch3QAAAnRJREFUII3syOXV84qCi/mgTqGR92Z0vO6V1R0bIHUeQf+j\np3+ApC8TpGfZtWyi8Zfe1/hqDntuOxtnnPp2Stk89l/j1HUTw5DE+a+x8KFdL4hz0+5+49R9\n7Mfl7JHQiep9QpDGNefy6jm/eJ8d+WM535lzhXFWuwZIhXdS/tB7S4CkLxnS+pZD+DkF64y9\nAR9yXnFik3X81DarOK88rn1d+YHFFZx/zP4agrSZi/N/Leyxm/OvsxvvbDi8aCvnW4oOr605\n7PA9nD9j7Gx4i63l8ur5m2wkW8n5EvaP0KXmZdRDWhe5OkPOAiR9TWEDxkVax8/er5T/3Lx/\n6C+9qOlV47qxezj/qkWbGx7skfES5y9mdr/vz606bOePsD/Pk86fwLqMva6gd+Pd37NyDhp7\nZ6fcOaHrnuwe40e263x86JoneyqXV88rW2aEzuZVnfa97ekr9mvTeU4E0itZaX+EPh0Bkr4a\ndn+zz58LP6ozgb1cw8Y8c3SLY58xvr/yvHaFJ4Tv47x3UsuiwaErlDV9m10lnz/19y2O+udb\ncR6QXdCvdeszFxlffvj7wlOXHXl+6NTZfzIgidXzoSx8kOXl/QrbX/jzfw44I3LRy/rp+enT\nPEDyttBfupurq5v0SejjzuY3hz7ObL47gdWXF+wFb0GvIUDyNpch8T4d5uxcfWGe8fYKdcc8\nlsDqnyhO/2PG6giQvM1tSD8fH7rlWDQrfHpxx3Lb1e/ptMjV+XttgORttVe/6fIaf3jvh/rj\n3C7aZrv67XDkToCEkAsBEkIuBEgIuRAgIeRCgISQCwESQi4ESAi5ECAh5EKAhJALARJCLvT/\nfITSJkbZ2+UAAAAASUVORK5CYII=",
      "text/plain": [
       "plot without title"
      ]
     },
     "metadata": {
      "image/png": {
       "height": 420,
       "width": 420
      }
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "i = 1\n",
    "print(traits[i])\n",
    "output = out_32to36_grm_regressed[[i]] # Trait 1 = traits[1]\n",
    "trait = traits[i]\n",
    "\n",
    "qq::qq_plot(output$results$p_value_REML.1) + ggtitle(paste0(\"GridLMM: Control - Main Effect - \", traits[i]))\n",
    "ggsave(paste0(\"plots/T32_GRM_qq_Ctrl_\",trait,\".png\"))\n",
    "gwasResults <- data.frame(BP = map_g$pos[1:length(output$results$p_value_REML.1)], CHR=chr[1:length(output$results$p_value_REML.1)], P = output$results$p_value_REML.1)\n",
    "gwasResults <- gwasResults[!is.na(gwasResults$P),]\n",
    "manhattanplot(gwasResults) + ggtitle(paste0(\"GridLMM: Control - Main Effect - \", traits[i]))\n",
    "ggsave(paste0(\"plots/T32_GRM_manh_Ctrl_\",trait,\".png\"))\n",
    "\n",
    "qq::qq_plot(output$results$p_value_REML.2) + ggtitle(paste0(\"GridLMM: GxE Effect - HS - \", traits[i]))\n",
    "ggsave(paste0(\"plots/T32_GRM_qq_GxE_HS_\",trait,\".png\"))\n",
    "gwasResults <- data.frame(BP = map_g$pos[1:length(output$results$p_value_REML.2)], CHR=chr[1:length(output$results$p_value_REML.2)], P = output$results$p_value_REML.2)\n",
    "gwasResults <- gwasResults[!is.na(gwasResults$P),]\n",
    "manhattanplot(gwasResults) + ggtitle(paste0(\"GridLMM: GxE Effect - HS - \", traits[i]))\n",
    "ggsave(paste0(\"plots/T32_GRM_manh_GxE_HS_\",trait,\".png\"))\n",
    "\n",
    "qq::qq_plot(output$results$p_value_REML.3) + ggtitle(paste0(\"GridLMM: GxE Effect - R - \", traits[i]))\n",
    "ggsave(paste0(\"plots/T32_GRM_qq_GxE_R_\",trait,\".png\"))\n",
    "gwasResults <- data.frame(BP = map_g$pos[1:length(output$results$p_value_REML.3)], CHR=chr[1:length(output$results$p_value_REML.3)], P = output$results$p_value_REML.3)\n",
    "gwasResults <- gwasResults[!is.na(gwasResults$P),]\n",
    "manhattanplot(gwasResults) + ggtitle(paste0(\"GridLMM: GxE Effect - R - \", traits[i]))\n",
    "ggsave(paste0(\"plots/T32_GRM_manh_GxE_R_\",trait,\".png\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "respiratory-malpractice",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "List of 4\n",
      " $ :List of 2\n",
      "  ..$ results:'data.frame':\t565448 obs. of  25 variables:\n",
      "  .. ..$ Trait                  : chr [1:565448] \"y\" \"y\" \"y\" \"y\" ...\n",
      "  .. ..$ X_ID                   : chr [1:565448] \"2L:5095\" \"2L:5233\" \"2L:5372\" \"2L:5904\" ...\n",
      "  .. ..$ s2                     : num [1:565448] 4262 4292 4279 4290 4300 ...\n",
      "  .. ..$ ML_logLik              : num [1:565448] -2128 -2129 -2129 -2129 -2128 ...\n",
      "  .. ..$ Family..Intercept..ML  : num [1:565448] 0.406 0.396 0.406 0.396 0.406 ...\n",
      "  .. ..$ Family.TreatmentHS.ML  : num [1:565448] 0 0 0 0 0 0 0 0 0 0 ...\n",
      "  .. ..$ Family.TreatmentR.ML   : num [1:565448] 0.0195 0.0195 0.0195 0.0195 0.0195 ...\n",
      "  .. ..$ beta.1                 : num [1:565448] 35.3 22.1 28.3 27 35.7 ...\n",
      "  .. ..$ beta.2                 : num [1:565448] -32.8 -19.01 -6.48 -33.42 -49.81 ...\n",
      "  .. ..$ beta.3                 : num [1:565448] -18.2 -21.8 -43.6 -20.3 -21.3 ...\n",
      "  .. ..$ beta.4                 : num [1:565448] -7.0401 3.4223 -0.0393 0.8516 -4.7884 ...\n",
      "  .. ..$ beta.5                 : num [1:565448] 5.41 -4.67 -11.28 4.34 14.38 ...\n",
      "  .. ..$ beta.6                 : num [1:565448] -3.7279 -0.0865 11.6886 -1.2308 -0.4443 ...\n",
      "  .. ..$ REML_logLik            : num [1:565448] -2098 -2099 -2099 -2099 -2098 ...\n",
      "  .. ..$ Family..Intercept..REML: num [1:565448] 0.396 0.396 0.396 0.396 0.406 ...\n",
      "  .. ..$ Family.TreatmentHS.REML: num [1:565448] 0 0 0 0 0 0 0 0 0 0 ...\n",
      "  .. ..$ Family.TreatmentR.REML : num [1:565448] 0.0195 0.0195 0.0195 0.0195 0.0195 ...\n",
      "  .. ..$ F.1                    : num [1:565448] 1.08 8.80e-02 5.35e-06 9.26e-03 2.80e-01 ...\n",
      "  .. ..$ F.2                    : num [1:565448] 0.5 0.136 0.361 0.191 2.074 ...\n",
      "  .. ..$ F.3                    : num [1:565448] 2.25e-01 4.39e-05 3.66e-01 1.46e-02 1.87e-03 ...\n",
      "  .. ..$ n_steps                : num [1:565448] 1 1 1 1 2 1 2 2 1 1 ...\n",
      "  .. ..$ Df_X                   : int [1:565448] 3 3 3 3 3 3 3 3 3 3 ...\n",
      "  .. ..$ p_value_REML.1         : num [1:565448] 0.3 0.767 0.998 0.923 0.597 ...\n",
      "  .. ..$ p_value_REML.2         : num [1:565448] 0.48 0.713 0.548 0.662 0.151 ...\n",
      "  .. ..$ p_value_REML.3         : num [1:565448] 0.635 0.995 0.545 0.904 0.966 ...\n",
      "  ..$ setup  :List of 8\n",
      "  .. ..$ Y               : num [1:387, 1] -17.3 -59.9 -40.5 76 31.3 ...\n",
      "  .. .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. .. ..$ : NULL\n",
      "  .. .. .. ..$ : chr \"y\"\n",
      "  .. ..$ X_cov           : num [1:387, 1:3] 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. .. ..$ : chr [1:387] \"1\" \"2\" \"3\" \"4\" ...\n",
      "  .. .. .. ..$ : chr [1:3] \"(Intercept)\" \"TreatmentHS\" \"TreatmentR\"\n",
      "  .. .. ..- attr(*, \"assign\")= int [1:3] 0 1 1\n",
      "  .. .. ..- attr(*, \"contrasts\")=List of 1\n",
      "  .. .. .. ..$ Treatment: chr \"contr.treatment\"\n",
      "  .. .. ..- attr(*, \"msgScaleX\")= chr(0) \n",
      "  .. ..$ h2_start        : num [1, 1:3] 0.3965 0 0.0195\n",
      "  .. .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. .. ..$ : NULL\n",
      "  .. .. .. ..$ : chr [1:3] \"Family..Intercept.\" \"Family.TreatmentHS\" \"Family.TreatmentR\"\n",
      "  .. ..$ V_setup         :List of 7\n",
      "  .. .. ..$ RE_setup            :List of 3\n",
      "  .. .. .. ..$ Family.(Intercept):List of 5\n",
      "  .. .. .. .. ..$ term  : chr \"Family\"\n",
      "  .. .. .. .. ..$ Z     :Formal class 'dgCMatrix' [package \"Matrix\"] with 6 slots\n",
      "  .. .. .. .. .. .. ..@ i       : int [1:387] 171 172 173 174 175 176 177 178 179 180 ...\n",
      "  .. .. .. .. .. .. ..@ p       : int [1:130] 0 3 6 9 12 15 18 21 24 27 ...\n",
      "  .. .. .. .. .. .. ..@ Dim     : int [1:2] 387 129\n",
      "  .. .. .. .. .. .. ..@ Dimnames:List of 2\n",
      "  .. .. .. .. .. .. .. ..$ : chr [1:387] \"1\" \"2\" \"3\" \"4\" ...\n",
      "  .. .. .. .. .. .. .. ..$ : chr [1:129] \"228\" \"233\" \"240\" \"244\" ...\n",
      "  .. .. .. .. .. .. ..@ x       : num [1:387] 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. .. .. .. .. .. ..@ factors : list()\n",
      "  .. .. .. .. ..$ K     :Formal class 'dsCMatrix' [package \"Matrix\"] with 7 slots\n",
      "  .. .. .. .. .. .. ..@ i       : int [1:8385] 0 0 1 0 1 2 0 1 2 3 ...\n",
      "  .. .. .. .. .. .. ..@ p       : int [1:130] 0 1 3 6 10 15 21 28 36 45 ...\n",
      "  .. .. .. .. .. .. ..@ Dim     : int [1:2] 129 129\n",
      "  .. .. .. .. .. .. ..@ Dimnames:List of 2\n",
      "  .. .. .. .. .. .. .. ..$ : Named chr [1:129] \"228\" \"233\" \"240\" \"244\" ...\n",
      "  .. .. .. .. .. .. .. .. ..- attr(*, \"names\")= chr [1:129] \"Plate1_10A_P_228_F\" \"Plate1_10B_P_233_F\" \"Plate1_10C_P_240_F\" \"Plate1_10D_P_244_F\" ...\n",
      "  .. .. .. .. .. .. .. ..$ : Named chr [1:129] \"228\" \"233\" \"240\" \"244\" ...\n",
      "  .. .. .. .. .. .. .. .. ..- attr(*, \"names\")= chr [1:129] \"Plate1_10A_P_228_F\" \"Plate1_10B_P_233_F\" \"Plate1_10C_P_240_F\" \"Plate1_10D_P_244_F\" ...\n",
      "  .. .. .. .. .. .. ..@ x       : num [1:8385] 1.0908 -0.1274 1.1131 0.0236 -0.0815 ...\n",
      "  .. .. .. .. .. .. ..@ uplo    : chr \"U\"\n",
      "  .. .. .. .. .. .. ..@ factors : list()\n",
      "  .. .. .. .. ..$ p     : NULL\n",
      "  .. .. .. .. ..$ p_test: NULL\n",
      "  .. .. .. ..$ Family.TreatmentHS:List of 5\n",
      "  .. .. .. .. ..$ term  : chr \"Family\"\n",
      "  .. .. .. .. ..$ Z     :Formal class 'dgCMatrix' [package \"Matrix\"] with 6 slots\n",
      "  .. .. .. .. .. .. ..@ i       : int [1:129] 172 175 178 181 184 187 190 193 196 199 ...\n",
      "  .. .. .. .. .. .. ..@ p       : int [1:130] 0 1 2 3 4 5 6 7 8 9 ...\n",
      "  .. .. .. .. .. .. ..@ Dim     : int [1:2] 387 129\n",
      "  .. .. .. .. .. .. ..@ Dimnames:List of 2\n",
      "  .. .. .. .. .. .. .. ..$ : chr [1:387] \"1\" \"2\" \"3\" \"4\" ...\n",
      "  .. .. .. .. .. .. .. ..$ : chr [1:129] \"228\" \"233\" \"240\" \"244\" ...\n",
      "  .. .. .. .. .. .. ..@ x       : num [1:129] 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. .. .. .. .. .. ..@ factors : list()\n",
      "  .. .. .. .. ..$ K     :Formal class 'dsCMatrix' [package \"Matrix\"] with 7 slots\n",
      "  .. .. .. .. .. .. ..@ i       : int [1:8385] 0 0 1 0 1 2 0 1 2 3 ...\n",
      "  .. .. .. .. .. .. ..@ p       : int [1:130] 0 1 3 6 10 15 21 28 36 45 ...\n",
      "  .. .. .. .. .. .. ..@ Dim     : int [1:2] 129 129\n",
      "  .. .. .. .. .. .. ..@ Dimnames:List of 2\n",
      "  .. .. .. .. .. .. .. ..$ : Named chr [1:129] \"228\" \"233\" \"240\" \"244\" ...\n",
      "  .. .. .. .. .. .. .. .. ..- attr(*, \"names\")= chr [1:129] \"Plate1_10A_P_228_F\" \"Plate1_10B_P_233_F\" \"Plate1_10C_P_240_F\" \"Plate1_10D_P_244_F\" ...\n",
      "  .. .. .. .. .. .. .. ..$ : Named chr [1:129] \"228\" \"233\" \"240\" \"244\" ...\n",
      "  .. .. .. .. .. .. .. .. ..- attr(*, \"names\")= chr [1:129] \"Plate1_10A_P_228_F\" \"Plate1_10B_P_233_F\" \"Plate1_10C_P_240_F\" \"Plate1_10D_P_244_F\" ...\n",
      "  .. .. .. .. .. .. ..@ x       : num [1:8385] 1.0908 -0.1274 1.1131 0.0236 -0.0815 ...\n",
      "  .. .. .. .. .. .. ..@ uplo    : chr \"U\"\n",
      "  .. .. .. .. .. .. ..@ factors : list()\n",
      "  .. .. .. .. ..$ p     : NULL\n",
      "  .. .. .. .. ..$ p_test: NULL\n",
      "  .. .. .. ..$ Family.TreatmentR :List of 5\n",
      "  .. .. .. .. ..$ term  : chr \"Family\"\n",
      "  .. .. .. .. ..$ Z     :Formal class 'dgCMatrix' [package \"Matrix\"] with 6 slots\n",
      "  .. .. .. .. .. .. ..@ i       : int [1:129] 173 176 179 182 185 188 191 194 197 200 ...\n",
      "  .. .. .. .. .. .. ..@ p       : int [1:130] 0 1 2 3 4 5 6 7 8 9 ...\n",
      "  .. .. .. .. .. .. ..@ Dim     : int [1:2] 387 129\n",
      "  .. .. .. .. .. .. ..@ Dimnames:List of 2\n",
      "  .. .. .. .. .. .. .. ..$ : chr [1:387] \"1\" \"2\" \"3\" \"4\" ...\n",
      "  .. .. .. .. .. .. .. ..$ : chr [1:129] \"228\" \"233\" \"240\" \"244\" ...\n",
      "  .. .. .. .. .. .. ..@ x       : num [1:129] 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. .. .. .. .. .. ..@ factors : list()\n",
      "  .. .. .. .. ..$ K     :Formal class 'dsCMatrix' [package \"Matrix\"] with 7 slots\n",
      "  .. .. .. .. .. .. ..@ i       : int [1:8385] 0 0 1 0 1 2 0 1 2 3 ...\n",
      "  .. .. .. .. .. .. ..@ p       : int [1:130] 0 1 3 6 10 15 21 28 36 45 ...\n",
      "  .. .. .. .. .. .. ..@ Dim     : int [1:2] 129 129\n",
      "  .. .. .. .. .. .. ..@ Dimnames:List of 2\n",
      "  .. .. .. .. .. .. .. ..$ : Named chr [1:129] \"228\" \"233\" \"240\" \"244\" ...\n",
      "  .. .. .. .. .. .. .. .. ..- attr(*, \"names\")= chr [1:129] \"Plate1_10A_P_228_F\" \"Plate1_10B_P_233_F\" \"Plate1_10C_P_240_F\" \"Plate1_10D_P_244_F\" ...\n",
      "  .. .. .. .. .. .. .. ..$ : Named chr [1:129] \"228\" \"233\" \"240\" \"244\" ...\n",
      "  .. .. .. .. .. .. .. .. ..- attr(*, \"names\")= chr [1:129] \"Plate1_10A_P_228_F\" \"Plate1_10B_P_233_F\" \"Plate1_10C_P_240_F\" \"Plate1_10D_P_244_F\" ...\n",
      "  .. .. .. .. .. .. ..@ x       : num [1:8385] 1.0908 -0.1274 1.1131 0.0236 -0.0815 ...\n",
      "  .. .. .. .. .. .. ..@ uplo    : chr \"U\"\n",
      "  .. .. .. .. .. .. ..@ factors : list()\n",
      "  .. .. .. .. ..$ p     : NULL\n",
      "  .. .. .. .. ..$ p_test: NULL\n",
      "  .. .. ..$ Qt                  : NULL\n",
      "  .. .. ..$ ZKZts               :List of 3\n",
      "  .. .. .. ..$ Family.(Intercept):Formal class 'dgCMatrix' [package \"Matrix\"] with 6 slots\n",
      "  .. .. .. .. .. ..@ i       : int [1:149769] 0 1 2 3 4 5 6 7 8 9 ...\n",
      "  .. .. .. .. .. ..@ p       : int [1:388] 0 387 774 1161 1548 1935 2322 2709 3096 3483 ...\n",
      "  .. .. .. .. .. ..@ Dim     : int [1:2] 387 387\n",
      "  .. .. .. .. .. ..@ Dimnames:List of 2\n",
      "  .. .. .. .. .. .. ..$ : chr [1:387] \"1\" \"2\" \"3\" \"4\" ...\n",
      "  .. .. .. .. .. .. ..$ : chr [1:387] \"1\" \"2\" \"3\" \"4\" ...\n",
      "  .. .. .. .. .. ..@ x       : num [1:149769] 1.007 1.007 1.007 0.0848 0.0848 ...\n",
      "  .. .. .. .. .. ..@ factors : list()\n",
      "  .. .. .. ..$ Family.TreatmentHS:Formal class 'dgCMatrix' [package \"Matrix\"] with 6 slots\n",
      "  .. .. .. .. .. ..@ i       : int [1:16641] 1 4 7 10 13 16 19 22 25 28 ...\n",
      "  .. .. .. .. .. ..@ p       : int [1:388] 0 0 129 129 129 258 258 258 387 387 ...\n",
      "  .. .. .. .. .. ..@ Dim     : int [1:2] 387 387\n",
      "  .. .. .. .. .. ..@ Dimnames:List of 2\n",
      "  .. .. .. .. .. .. ..$ : chr [1:387] \"1\" \"2\" \"3\" \"4\" ...\n",
      "  .. .. .. .. .. .. ..$ : chr [1:387] \"1\" \"2\" \"3\" \"4\" ...\n",
      "  .. .. .. .. .. ..@ x       : num [1:16641] 3.021 0.2543 -0.1691 -0.1904 0.0291 ...\n",
      "  .. .. .. .. .. ..@ factors : list()\n",
      "  .. .. .. ..$ Family.TreatmentR :Formal class 'dgCMatrix' [package \"Matrix\"] with 6 slots\n",
      "  .. .. .. .. .. ..@ i       : int [1:16641] 2 5 8 11 14 17 20 23 26 29 ...\n",
      "  .. .. .. .. .. ..@ p       : int [1:388] 0 0 0 129 129 129 258 258 258 387 ...\n",
      "  .. .. .. .. .. ..@ Dim     : int [1:2] 387 387\n",
      "  .. .. .. .. .. ..@ Dimnames:List of 2\n",
      "  .. .. .. .. .. .. ..$ : chr [1:387] \"1\" \"2\" \"3\" \"4\" ...\n",
      "  .. .. .. .. .. .. ..$ : chr [1:387] \"1\" \"2\" \"3\" \"4\" ...\n",
      "  .. .. .. .. .. ..@ x       : num [1:16641] 3.021 0.2543 -0.1691 -0.1904 0.0291 ...\n",
      "  .. .. .. .. .. ..@ factors : list()\n",
      "  .. .. ..$ resid_V             : num [1:387, 1:387] 1 0 0 0 0 0 0 0 0 0 ...\n",
      "  .. .. ..$ save_V_folder       : NULL\n",
      "  .. .. ..$ n_SNPs_downdated_RRM: Named num [1:3] Inf Inf Inf\n",
      "  .. .. .. ..- attr(*, \"names\")= chr [1:3] \"Family.(Intercept)\" \"Family.TreatmentHS\" \"Family.TreatmentR\"\n",
      "  .. .. ..$ downdate_ratios     : Named num [1:3] 1 1 1\n",
      "  .. .. .. ..- attr(*, \"names\")= chr [1:3] \"Family.(Intercept)\" \"Family.TreatmentHS\" \"Family.TreatmentR\"\n",
      "  .. ..$ mm_test         : num [1:387, 1:3] 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. .. ..$ : chr [1:387] \"1\" \"2\" \"3\" \"4\" ...\n",
      "  .. .. .. ..$ : chr [1:3] \"(Intercept)\" \"TreatmentHS\" \"TreatmentR\"\n",
      "  .. .. ..- attr(*, \"assign\")= int [1:3] 0 1 1\n",
      "  .. .. ..- attr(*, \"contrasts\")=List of 1\n",
      "  .. .. .. ..$ Treatment: chr \"contr.treatment\"\n",
      "  .. ..$ mm_reduced      : num [1:387, 1] 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. .. ..$ : chr [1:387] \"1\" \"2\" \"3\" \"4\" ...\n",
      "  .. .. .. ..$ : chr \"(Intercept)\"\n",
      "  .. .. ..- attr(*, \"assign\")= int 0\n",
      "  .. ..$ proximal_markers: NULL\n",
      "  .. ..$ proximal_Xs     : NULL\n",
      " $ :List of 2\n",
      "  ..$ results:'data.frame':\t565448 obs. of  25 variables:\n",
      "  .. ..$ Trait                  : chr [1:565448] \"y\" \"y\" \"y\" \"y\" ...\n",
      "  .. ..$ X_ID                   : chr [1:565448] \"2L:5095\" \"2L:5233\" \"2L:5372\" \"2L:5904\" ...\n",
      "  .. ..$ s2                     : num [1:565448] 380 382 383 384 382 ...\n",
      "  .. ..$ ML_logLik              : num [1:565448] -1660 -1662 -1662 -1663 -1662 ...\n",
      "  .. ..$ Family..Intercept..ML  : num [1:565448] 0.389 0.379 0.379 0.379 0.379 ...\n",
      "  .. ..$ Family.TreatmentHS.ML  : num [1:565448] 0 0 0 0 0 0 0 0 0 0 ...\n",
      "  .. ..$ Family.TreatmentR.ML   : num [1:565448] 0.0747 0.0647 0.0647 0.0647 0.0647 ...\n",
      "  .. ..$ beta.1                 : num [1:565448] -19.5 -24 -31.3 -20.4 -21.9 ...\n",
      "  .. ..$ beta.2                 : num [1:565448] -2.313 -2.533 13.928 1.285 0.533 ...\n",
      "  .. ..$ beta.3                 : num [1:565448] 16.6 23.5 30.2 18.6 16.1 ...\n",
      "  .. ..$ beta.4                 : num [1:565448] -0.9967 1.9784 5.8287 -0.0665 0.8996 ...\n",
      "  .. ..$ beta.5                 : num [1:565448] 4.188 2.49 -6.494 0.449 0.884 ...\n",
      "  .. ..$ beta.6                 : num [1:565448] 3.89 -1.71 -5.24 1.36 2.81 ...\n",
      "  .. ..$ REML_logLik            : num [1:565448] -1638 -1640 -1640 -1641 -1640 ...\n",
      "  .. ..$ Family..Intercept..REML: num [1:565448] 0.379 0.379 0.379 0.379 0.379 ...\n",
      "  .. ..$ Family.TreatmentHS.REML: num [1:565448] 0 0 0 0 0 0 0 0 0 0 ...\n",
      "  .. ..$ Family.TreatmentR.REML : num [1:565448] 0.0647 0.0547 0.0647 0.0547 0.0547 ...\n",
      "  .. ..$ F.1                    : num [1:565448] 0.253061 0.342072 1.37701 0.000654 0.114932 ...\n",
      "  .. ..$ F.2                    : num [1:565448] 3.527 0.4474 1.4043 0.0236 0.0893 ...\n",
      "  .. ..$ F.3                    : num [1:565448] 2.582 0.181 0.765 0.188 0.78 ...\n",
      "  .. ..$ n_steps                : num [1:565448] 2 1 2 1 1 2 3 3 1 1 ...\n",
      "  .. ..$ Df_X                   : int [1:565448] 3 3 3 3 3 3 3 3 3 3 ...\n",
      "  .. ..$ p_value_REML.1         : num [1:565448] 0.615 0.559 0.241 0.98 0.735 ...\n",
      "  .. ..$ p_value_REML.2         : num [1:565448] 0.0611 0.504 0.2367 0.878 0.7652 ...\n",
      "  .. ..$ p_value_REML.3         : num [1:565448] 0.109 0.671 0.382 0.664 0.378 ...\n",
      "  ..$ setup  :List of 8\n",
      "  .. ..$ Y               : num [1:387, 1] -23.75 0.792 20.183 28.498 25.549 ...\n",
      "  .. .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. .. ..$ : NULL\n",
      "  .. .. .. ..$ : chr \"y\"\n",
      "  .. ..$ X_cov           : num [1:387, 1:3] 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. .. ..$ : chr [1:387] \"1\" \"2\" \"3\" \"4\" ...\n",
      "  .. .. .. ..$ : chr [1:3] \"(Intercept)\" \"TreatmentHS\" \"TreatmentR\"\n",
      "  .. .. ..- attr(*, \"assign\")= int [1:3] 0 1 1\n",
      "  .. .. ..- attr(*, \"contrasts\")=List of 1\n",
      "  .. .. .. ..$ Treatment: chr \"contr.treatment\"\n",
      "  .. .. ..- attr(*, \"msgScaleX\")= chr(0) \n",
      "  .. ..$ h2_start        : num [1, 1:3] 0.3789 0 0.0547\n",
      "  .. .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. .. ..$ : NULL\n",
      "  .. .. .. ..$ : chr [1:3] \"Family..Intercept.\" \"Family.TreatmentHS\" \"Family.TreatmentR\"\n",
      "  .. ..$ V_setup         :List of 7\n",
      "  .. .. ..$ RE_setup            :List of 3\n",
      "  .. .. .. ..$ Family.(Intercept):List of 5\n",
      "  .. .. .. .. ..$ term  : chr \"Family\"\n",
      "  .. .. .. .. ..$ Z     :Formal class 'dgCMatrix' [package \"Matrix\"] with 6 slots\n",
      "  .. .. .. .. .. .. ..@ i       : int [1:387] 171 172 173 174 175 176 177 178 179 180 ...\n",
      "  .. .. .. .. .. .. ..@ p       : int [1:130] 0 3 6 9 12 15 18 21 24 27 ...\n",
      "  .. .. .. .. .. .. ..@ Dim     : int [1:2] 387 129\n",
      "  .. .. .. .. .. .. ..@ Dimnames:List of 2\n",
      "  .. .. .. .. .. .. .. ..$ : chr [1:387] \"1\" \"2\" \"3\" \"4\" ...\n",
      "  .. .. .. .. .. .. .. ..$ : chr [1:129] \"228\" \"233\" \"240\" \"244\" ...\n",
      "  .. .. .. .. .. .. ..@ x       : num [1:387] 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. .. .. .. .. .. ..@ factors : list()\n",
      "  .. .. .. .. ..$ K     :Formal class 'dsCMatrix' [package \"Matrix\"] with 7 slots\n",
      "  .. .. .. .. .. .. ..@ i       : int [1:8385] 0 0 1 0 1 2 0 1 2 3 ...\n",
      "  .. .. .. .. .. .. ..@ p       : int [1:130] 0 1 3 6 10 15 21 28 36 45 ...\n",
      "  .. .. .. .. .. .. ..@ Dim     : int [1:2] 129 129\n",
      "  .. .. .. .. .. .. ..@ Dimnames:List of 2\n",
      "  .. .. .. .. .. .. .. ..$ : Named chr [1:129] \"228\" \"233\" \"240\" \"244\" ...\n",
      "  .. .. .. .. .. .. .. .. ..- attr(*, \"names\")= chr [1:129] \"Plate1_10A_P_228_F\" \"Plate1_10B_P_233_F\" \"Plate1_10C_P_240_F\" \"Plate1_10D_P_244_F\" ...\n",
      "  .. .. .. .. .. .. .. ..$ : Named chr [1:129] \"228\" \"233\" \"240\" \"244\" ...\n",
      "  .. .. .. .. .. .. .. .. ..- attr(*, \"names\")= chr [1:129] \"Plate1_10A_P_228_F\" \"Plate1_10B_P_233_F\" \"Plate1_10C_P_240_F\" \"Plate1_10D_P_244_F\" ...\n",
      "  .. .. .. .. .. .. ..@ x       : num [1:8385] 1.0908 -0.1274 1.1131 0.0236 -0.0815 ...\n",
      "  .. .. .. .. .. .. ..@ uplo    : chr \"U\"\n",
      "  .. .. .. .. .. .. ..@ factors : list()\n",
      "  .. .. .. .. ..$ p     : NULL\n",
      "  .. .. .. .. ..$ p_test: NULL\n",
      "  .. .. .. ..$ Family.TreatmentHS:List of 5\n",
      "  .. .. .. .. ..$ term  : chr \"Family\"\n",
      "  .. .. .. .. ..$ Z     :Formal class 'dgCMatrix' [package \"Matrix\"] with 6 slots\n",
      "  .. .. .. .. .. .. ..@ i       : int [1:129] 172 175 178 181 184 187 190 193 196 199 ...\n",
      "  .. .. .. .. .. .. ..@ p       : int [1:130] 0 1 2 3 4 5 6 7 8 9 ...\n",
      "  .. .. .. .. .. .. ..@ Dim     : int [1:2] 387 129\n",
      "  .. .. .. .. .. .. ..@ Dimnames:List of 2\n",
      "  .. .. .. .. .. .. .. ..$ : chr [1:387] \"1\" \"2\" \"3\" \"4\" ...\n",
      "  .. .. .. .. .. .. .. ..$ : chr [1:129] \"228\" \"233\" \"240\" \"244\" ...\n",
      "  .. .. .. .. .. .. ..@ x       : num [1:129] 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. .. .. .. .. .. ..@ factors : list()\n",
      "  .. .. .. .. ..$ K     :Formal class 'dsCMatrix' [package \"Matrix\"] with 7 slots\n",
      "  .. .. .. .. .. .. ..@ i       : int [1:8385] 0 0 1 0 1 2 0 1 2 3 ...\n",
      "  .. .. .. .. .. .. ..@ p       : int [1:130] 0 1 3 6 10 15 21 28 36 45 ...\n",
      "  .. .. .. .. .. .. ..@ Dim     : int [1:2] 129 129\n",
      "  .. .. .. .. .. .. ..@ Dimnames:List of 2\n",
      "  .. .. .. .. .. .. .. ..$ : Named chr [1:129] \"228\" \"233\" \"240\" \"244\" ...\n",
      "  .. .. .. .. .. .. .. .. ..- attr(*, \"names\")= chr [1:129] \"Plate1_10A_P_228_F\" \"Plate1_10B_P_233_F\" \"Plate1_10C_P_240_F\" \"Plate1_10D_P_244_F\" ...\n",
      "  .. .. .. .. .. .. .. ..$ : Named chr [1:129] \"228\" \"233\" \"240\" \"244\" ...\n",
      "  .. .. .. .. .. .. .. .. ..- attr(*, \"names\")= chr [1:129] \"Plate1_10A_P_228_F\" \"Plate1_10B_P_233_F\" \"Plate1_10C_P_240_F\" \"Plate1_10D_P_244_F\" ...\n",
      "  .. .. .. .. .. .. ..@ x       : num [1:8385] 1.0908 -0.1274 1.1131 0.0236 -0.0815 ...\n",
      "  .. .. .. .. .. .. ..@ uplo    : chr \"U\"\n",
      "  .. .. .. .. .. .. ..@ factors : list()\n",
      "  .. .. .. .. ..$ p     : NULL\n",
      "  .. .. .. .. ..$ p_test: NULL\n",
      "  .. .. .. ..$ Family.TreatmentR :List of 5\n",
      "  .. .. .. .. ..$ term  : chr \"Family\"\n",
      "  .. .. .. .. ..$ Z     :Formal class 'dgCMatrix' [package \"Matrix\"] with 6 slots\n",
      "  .. .. .. .. .. .. ..@ i       : int [1:129] 173 176 179 182 185 188 191 194 197 200 ...\n",
      "  .. .. .. .. .. .. ..@ p       : int [1:130] 0 1 2 3 4 5 6 7 8 9 ...\n",
      "  .. .. .. .. .. .. ..@ Dim     : int [1:2] 387 129\n",
      "  .. .. .. .. .. .. ..@ Dimnames:List of 2\n",
      "  .. .. .. .. .. .. .. ..$ : chr [1:387] \"1\" \"2\" \"3\" \"4\" ...\n",
      "  .. .. .. .. .. .. .. ..$ : chr [1:129] \"228\" \"233\" \"240\" \"244\" ...\n",
      "  .. .. .. .. .. .. ..@ x       : num [1:129] 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. .. .. .. .. .. ..@ factors : list()\n",
      "  .. .. .. .. ..$ K     :Formal class 'dsCMatrix' [package \"Matrix\"] with 7 slots\n",
      "  .. .. .. .. .. .. ..@ i       : int [1:8385] 0 0 1 0 1 2 0 1 2 3 ...\n",
      "  .. .. .. .. .. .. ..@ p       : int [1:130] 0 1 3 6 10 15 21 28 36 45 ...\n",
      "  .. .. .. .. .. .. ..@ Dim     : int [1:2] 129 129\n",
      "  .. .. .. .. .. .. ..@ Dimnames:List of 2\n",
      "  .. .. .. .. .. .. .. ..$ : Named chr [1:129] \"228\" \"233\" \"240\" \"244\" ...\n",
      "  .. .. .. .. .. .. .. .. ..- attr(*, \"names\")= chr [1:129] \"Plate1_10A_P_228_F\" \"Plate1_10B_P_233_F\" \"Plate1_10C_P_240_F\" \"Plate1_10D_P_244_F\" ...\n",
      "  .. .. .. .. .. .. .. ..$ : Named chr [1:129] \"228\" \"233\" \"240\" \"244\" ...\n",
      "  .. .. .. .. .. .. .. .. ..- attr(*, \"names\")= chr [1:129] \"Plate1_10A_P_228_F\" \"Plate1_10B_P_233_F\" \"Plate1_10C_P_240_F\" \"Plate1_10D_P_244_F\" ...\n",
      "  .. .. .. .. .. .. ..@ x       : num [1:8385] 1.0908 -0.1274 1.1131 0.0236 -0.0815 ...\n",
      "  .. .. .. .. .. .. ..@ uplo    : chr \"U\"\n",
      "  .. .. .. .. .. .. ..@ factors : list()\n",
      "  .. .. .. .. ..$ p     : NULL\n",
      "  .. .. .. .. ..$ p_test: NULL\n",
      "  .. .. ..$ Qt                  : NULL\n",
      "  .. .. ..$ ZKZts               :List of 3\n",
      "  .. .. .. ..$ Family.(Intercept):Formal class 'dgCMatrix' [package \"Matrix\"] with 6 slots\n",
      "  .. .. .. .. .. ..@ i       : int [1:149769] 0 1 2 3 4 5 6 7 8 9 ...\n",
      "  .. .. .. .. .. ..@ p       : int [1:388] 0 387 774 1161 1548 1935 2322 2709 3096 3483 ...\n",
      "  .. .. .. .. .. ..@ Dim     : int [1:2] 387 387\n",
      "  .. .. .. .. .. ..@ Dimnames:List of 2\n",
      "  .. .. .. .. .. .. ..$ : chr [1:387] \"1\" \"2\" \"3\" \"4\" ...\n",
      "  .. .. .. .. .. .. ..$ : chr [1:387] \"1\" \"2\" \"3\" \"4\" ...\n",
      "  .. .. .. .. .. ..@ x       : num [1:149769] 1.007 1.007 1.007 0.0848 0.0848 ...\n",
      "  .. .. .. .. .. ..@ factors : list()\n",
      "  .. .. .. ..$ Family.TreatmentHS:Formal class 'dgCMatrix' [package \"Matrix\"] with 6 slots\n",
      "  .. .. .. .. .. ..@ i       : int [1:16641] 1 4 7 10 13 16 19 22 25 28 ...\n",
      "  .. .. .. .. .. ..@ p       : int [1:388] 0 0 129 129 129 258 258 258 387 387 ...\n",
      "  .. .. .. .. .. ..@ Dim     : int [1:2] 387 387\n",
      "  .. .. .. .. .. ..@ Dimnames:List of 2\n",
      "  .. .. .. .. .. .. ..$ : chr [1:387] \"1\" \"2\" \"3\" \"4\" ...\n",
      "  .. .. .. .. .. .. ..$ : chr [1:387] \"1\" \"2\" \"3\" \"4\" ...\n",
      "  .. .. .. .. .. ..@ x       : num [1:16641] 3.021 0.2543 -0.1691 -0.1904 0.0291 ...\n",
      "  .. .. .. .. .. ..@ factors : list()\n",
      "  .. .. .. ..$ Family.TreatmentR :Formal class 'dgCMatrix' [package \"Matrix\"] with 6 slots\n",
      "  .. .. .. .. .. ..@ i       : int [1:16641] 2 5 8 11 14 17 20 23 26 29 ...\n",
      "  .. .. .. .. .. ..@ p       : int [1:388] 0 0 0 129 129 129 258 258 258 387 ...\n",
      "  .. .. .. .. .. ..@ Dim     : int [1:2] 387 387\n",
      "  .. .. .. .. .. ..@ Dimnames:List of 2\n",
      "  .. .. .. .. .. .. ..$ : chr [1:387] \"1\" \"2\" \"3\" \"4\" ...\n",
      "  .. .. .. .. .. .. ..$ : chr [1:387] \"1\" \"2\" \"3\" \"4\" ...\n",
      "  .. .. .. .. .. ..@ x       : num [1:16641] 3.021 0.2543 -0.1691 -0.1904 0.0291 ...\n",
      "  .. .. .. .. .. ..@ factors : list()\n",
      "  .. .. ..$ resid_V             : num [1:387, 1:387] 1 0 0 0 0 0 0 0 0 0 ...\n",
      "  .. .. ..$ save_V_folder       : NULL\n",
      "  .. .. ..$ n_SNPs_downdated_RRM: Named num [1:3] Inf Inf Inf\n",
      "  .. .. .. ..- attr(*, \"names\")= chr [1:3] \"Family.(Intercept)\" \"Family.TreatmentHS\" \"Family.TreatmentR\"\n",
      "  .. .. ..$ downdate_ratios     : Named num [1:3] 1 1 1\n",
      "  .. .. .. ..- attr(*, \"names\")= chr [1:3] \"Family.(Intercept)\" \"Family.TreatmentHS\" \"Family.TreatmentR\"\n",
      "  .. ..$ mm_test         : num [1:387, 1:3] 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. .. ..$ : chr [1:387] \"1\" \"2\" \"3\" \"4\" ...\n",
      "  .. .. .. ..$ : chr [1:3] \"(Intercept)\" \"TreatmentHS\" \"TreatmentR\"\n",
      "  .. .. ..- attr(*, \"assign\")= int [1:3] 0 1 1\n",
      "  .. .. ..- attr(*, \"contrasts\")=List of 1\n",
      "  .. .. .. ..$ Treatment: chr \"contr.treatment\"\n",
      "  .. ..$ mm_reduced      : num [1:387, 1] 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. .. ..$ : chr [1:387] \"1\" \"2\" \"3\" \"4\" ...\n",
      "  .. .. .. ..$ : chr \"(Intercept)\"\n",
      "  .. .. ..- attr(*, \"assign\")= int 0\n",
      "  .. ..$ proximal_markers: NULL\n",
      "  .. ..$ proximal_Xs     : NULL\n",
      " $ :List of 2\n",
      "  ..$ results:'data.frame':\t565448 obs. of  25 variables:\n",
      "  .. ..$ Trait                  : chr [1:565448] \"y\" \"y\" \"y\" \"y\" ...\n",
      "  .. ..$ X_ID                   : chr [1:565448] \"2L:5095\" \"2L:5233\" \"2L:5372\" \"2L:5904\" ...\n",
      "  .. ..$ s2                     : num [1:565448] 4.99 5.03 4.93 5.03 5.06 ...\n",
      "  .. ..$ ML_logLik              : num [1:565448] -801 -805 -803 -805 -804 ...\n",
      "  .. ..$ Family..Intercept..ML  : num [1:565448] 0.518 0.498 0.498 0.498 0.508 ...\n",
      "  .. ..$ Family.TreatmentHS.ML  : num [1:565448] 0 0 0 0 0 0 0 0 0 0 ...\n",
      "  .. ..$ Family.TreatmentR.ML   : num [1:565448] 0.00586 0.00586 0.00586 0.00586 0.00586 ...\n",
      "  .. ..$ beta.1                 : num [1:565448] -1.17 -1.35 -3.28 -1.34 -1.01 ...\n",
      "  .. ..$ beta.2                 : num [1:565448] -1.634 -1.536 0.125 -1.354 -1.852 ...\n",
      "  .. ..$ beta.3                 : num [1:565448] -0.54 -0.613 -0.591 -0.789 -0.968 ...\n",
      "  .. ..$ beta.4                 : num [1:565448] -0.3127 -0.0747 0.9735 -0.1024 -0.3028 ...\n",
      "  .. ..$ beta.5                 : num [1:565448] 0.639 0.306 -0.602 0.26 0.552 ...\n",
      "  .. ..$ beta.6                 : num [1:565448] 0.0904 0.0919 0.0767 0.2425 0.332 ...\n",
      "  .. ..$ REML_logLik            : num [1:565448] -792 -796 -794 -796 -795 ...\n",
      "  .. ..$ Family..Intercept..REML: num [1:565448] 0.508 0.498 0.488 0.498 0.508 ...\n",
      "  .. ..$ Family.TreatmentHS.REML: num [1:565448] 0 0 0 0 0 0 0 0 0 0 ...\n",
      "  .. ..$ Family.TreatmentR.REML : num [1:565448] 0.00586 0.00586 0.00586 0.00586 0.00586 ...\n",
      "  .. ..$ F.1                    : num [1:565448] 1.8504 0.0359 2.8695 0.1159 0.9643 ...\n",
      "  .. ..$ F.2                    : num [1:565448] 7.161 0.587 1.032 0.689 3.069 ...\n",
      "  .. ..$ F.3                    : num [1:565448] 0.1404 0.0517 0.0164 0.588 1.0859 ...\n",
      "  .. ..$ n_steps                : num [1:565448] 2 1 2 1 2 1 2 1 1 1 ...\n",
      "  .. ..$ Df_X                   : int [1:565448] 3 3 3 3 3 3 3 3 3 3 ...\n",
      "  .. ..$ p_value_REML.1         : num [1:565448] 0.1745 0.8498 0.0911 0.7337 0.3267 ...\n",
      "  .. ..$ p_value_REML.2         : num [1:565448] 0.00777 0.4442 0.3104 0.40705 0.0806 ...\n",
      "  .. ..$ p_value_REML.3         : num [1:565448] 0.708 0.82 0.898 0.444 0.298 ...\n",
      "  ..$ setup  :List of 8\n",
      "  .. ..$ Y               : num [1:387, 1] -0.605 -1.71 -0.199 2.976 -0.354 ...\n",
      "  .. .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. .. ..$ : NULL\n",
      "  .. .. .. ..$ : chr \"y\"\n",
      "  .. ..$ X_cov           : num [1:387, 1:3] 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. .. ..$ : chr [1:387] \"1\" \"2\" \"3\" \"4\" ...\n",
      "  .. .. .. ..$ : chr [1:3] \"(Intercept)\" \"TreatmentHS\" \"TreatmentR\"\n",
      "  .. .. ..- attr(*, \"assign\")= int [1:3] 0 1 1\n",
      "  .. .. ..- attr(*, \"contrasts\")=List of 1\n",
      "  .. .. .. ..$ Treatment: chr \"contr.treatment\"\n",
      "  .. .. ..- attr(*, \"msgScaleX\")= chr(0) \n",
      "  .. ..$ h2_start        : num [1, 1:3] 0.49805 0 0.00586\n",
      "  .. .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. .. ..$ : NULL\n",
      "  .. .. .. ..$ : chr [1:3] \"Family..Intercept.\" \"Family.TreatmentHS\" \"Family.TreatmentR\"\n",
      "  .. ..$ V_setup         :List of 7\n",
      "  .. .. ..$ RE_setup            :List of 3\n",
      "  .. .. .. ..$ Family.(Intercept):List of 5\n",
      "  .. .. .. .. ..$ term  : chr \"Family\"\n",
      "  .. .. .. .. ..$ Z     :Formal class 'dgCMatrix' [package \"Matrix\"] with 6 slots\n",
      "  .. .. .. .. .. .. ..@ i       : int [1:387] 171 172 173 174 175 176 177 178 179 180 ...\n",
      "  .. .. .. .. .. .. ..@ p       : int [1:130] 0 3 6 9 12 15 18 21 24 27 ...\n",
      "  .. .. .. .. .. .. ..@ Dim     : int [1:2] 387 129\n",
      "  .. .. .. .. .. .. ..@ Dimnames:List of 2\n",
      "  .. .. .. .. .. .. .. ..$ : chr [1:387] \"1\" \"2\" \"3\" \"4\" ...\n",
      "  .. .. .. .. .. .. .. ..$ : chr [1:129] \"228\" \"233\" \"240\" \"244\" ...\n",
      "  .. .. .. .. .. .. ..@ x       : num [1:387] 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. .. .. .. .. .. ..@ factors : list()\n",
      "  .. .. .. .. ..$ K     :Formal class 'dsCMatrix' [package \"Matrix\"] with 7 slots\n",
      "  .. .. .. .. .. .. ..@ i       : int [1:8385] 0 0 1 0 1 2 0 1 2 3 ...\n",
      "  .. .. .. .. .. .. ..@ p       : int [1:130] 0 1 3 6 10 15 21 28 36 45 ...\n",
      "  .. .. .. .. .. .. ..@ Dim     : int [1:2] 129 129\n",
      "  .. .. .. .. .. .. ..@ Dimnames:List of 2\n",
      "  .. .. .. .. .. .. .. ..$ : Named chr [1:129] \"228\" \"233\" \"240\" \"244\" ...\n",
      "  .. .. .. .. .. .. .. .. ..- attr(*, \"names\")= chr [1:129] \"Plate1_10A_P_228_F\" \"Plate1_10B_P_233_F\" \"Plate1_10C_P_240_F\" \"Plate1_10D_P_244_F\" ...\n",
      "  .. .. .. .. .. .. .. ..$ : Named chr [1:129] \"228\" \"233\" \"240\" \"244\" ...\n",
      "  .. .. .. .. .. .. .. .. ..- attr(*, \"names\")= chr [1:129] \"Plate1_10A_P_228_F\" \"Plate1_10B_P_233_F\" \"Plate1_10C_P_240_F\" \"Plate1_10D_P_244_F\" ...\n",
      "  .. .. .. .. .. .. ..@ x       : num [1:8385] 1.0908 -0.1274 1.1131 0.0236 -0.0815 ...\n",
      "  .. .. .. .. .. .. ..@ uplo    : chr \"U\"\n",
      "  .. .. .. .. .. .. ..@ factors : list()\n",
      "  .. .. .. .. ..$ p     : NULL\n",
      "  .. .. .. .. ..$ p_test: NULL\n",
      "  .. .. .. ..$ Family.TreatmentHS:List of 5\n",
      "  .. .. .. .. ..$ term  : chr \"Family\"\n",
      "  .. .. .. .. ..$ Z     :Formal class 'dgCMatrix' [package \"Matrix\"] with 6 slots\n",
      "  .. .. .. .. .. .. ..@ i       : int [1:129] 172 175 178 181 184 187 190 193 196 199 ...\n",
      "  .. .. .. .. .. .. ..@ p       : int [1:130] 0 1 2 3 4 5 6 7 8 9 ...\n",
      "  .. .. .. .. .. .. ..@ Dim     : int [1:2] 387 129\n",
      "  .. .. .. .. .. .. ..@ Dimnames:List of 2\n",
      "  .. .. .. .. .. .. .. ..$ : chr [1:387] \"1\" \"2\" \"3\" \"4\" ...\n",
      "  .. .. .. .. .. .. .. ..$ : chr [1:129] \"228\" \"233\" \"240\" \"244\" ...\n",
      "  .. .. .. .. .. .. ..@ x       : num [1:129] 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. .. .. .. .. .. ..@ factors : list()\n",
      "  .. .. .. .. ..$ K     :Formal class 'dsCMatrix' [package \"Matrix\"] with 7 slots\n",
      "  .. .. .. .. .. .. ..@ i       : int [1:8385] 0 0 1 0 1 2 0 1 2 3 ...\n",
      "  .. .. .. .. .. .. ..@ p       : int [1:130] 0 1 3 6 10 15 21 28 36 45 ...\n",
      "  .. .. .. .. .. .. ..@ Dim     : int [1:2] 129 129\n",
      "  .. .. .. .. .. .. ..@ Dimnames:List of 2\n",
      "  .. .. .. .. .. .. .. ..$ : Named chr [1:129] \"228\" \"233\" \"240\" \"244\" ...\n",
      "  .. .. .. .. .. .. .. .. ..- attr(*, \"names\")= chr [1:129] \"Plate1_10A_P_228_F\" \"Plate1_10B_P_233_F\" \"Plate1_10C_P_240_F\" \"Plate1_10D_P_244_F\" ...\n",
      "  .. .. .. .. .. .. .. ..$ : Named chr [1:129] \"228\" \"233\" \"240\" \"244\" ...\n",
      "  .. .. .. .. .. .. .. .. ..- attr(*, \"names\")= chr [1:129] \"Plate1_10A_P_228_F\" \"Plate1_10B_P_233_F\" \"Plate1_10C_P_240_F\" \"Plate1_10D_P_244_F\" ...\n",
      "  .. .. .. .. .. .. ..@ x       : num [1:8385] 1.0908 -0.1274 1.1131 0.0236 -0.0815 ...\n",
      "  .. .. .. .. .. .. ..@ uplo    : chr \"U\"\n",
      "  .. .. .. .. .. .. ..@ factors : list()\n",
      "  .. .. .. .. ..$ p     : NULL\n",
      "  .. .. .. .. ..$ p_test: NULL\n",
      "  .. .. .. ..$ Family.TreatmentR :List of 5\n",
      "  .. .. .. .. ..$ term  : chr \"Family\"\n",
      "  .. .. .. .. ..$ Z     :Formal class 'dgCMatrix' [package \"Matrix\"] with 6 slots\n",
      "  .. .. .. .. .. .. ..@ i       : int [1:129] 173 176 179 182 185 188 191 194 197 200 ...\n",
      "  .. .. .. .. .. .. ..@ p       : int [1:130] 0 1 2 3 4 5 6 7 8 9 ...\n",
      "  .. .. .. .. .. .. ..@ Dim     : int [1:2] 387 129\n",
      "  .. .. .. .. .. .. ..@ Dimnames:List of 2\n",
      "  .. .. .. .. .. .. .. ..$ : chr [1:387] \"1\" \"2\" \"3\" \"4\" ...\n",
      "  .. .. .. .. .. .. .. ..$ : chr [1:129] \"228\" \"233\" \"240\" \"244\" ...\n",
      "  .. .. .. .. .. .. ..@ x       : num [1:129] 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. .. .. .. .. .. ..@ factors : list()\n",
      "  .. .. .. .. ..$ K     :Formal class 'dsCMatrix' [package \"Matrix\"] with 7 slots\n",
      "  .. .. .. .. .. .. ..@ i       : int [1:8385] 0 0 1 0 1 2 0 1 2 3 ...\n",
      "  .. .. .. .. .. .. ..@ p       : int [1:130] 0 1 3 6 10 15 21 28 36 45 ...\n",
      "  .. .. .. .. .. .. ..@ Dim     : int [1:2] 129 129\n",
      "  .. .. .. .. .. .. ..@ Dimnames:List of 2\n",
      "  .. .. .. .. .. .. .. ..$ : Named chr [1:129] \"228\" \"233\" \"240\" \"244\" ...\n",
      "  .. .. .. .. .. .. .. .. ..- attr(*, \"names\")= chr [1:129] \"Plate1_10A_P_228_F\" \"Plate1_10B_P_233_F\" \"Plate1_10C_P_240_F\" \"Plate1_10D_P_244_F\" ...\n",
      "  .. .. .. .. .. .. .. ..$ : Named chr [1:129] \"228\" \"233\" \"240\" \"244\" ...\n",
      "  .. .. .. .. .. .. .. .. ..- attr(*, \"names\")= chr [1:129] \"Plate1_10A_P_228_F\" \"Plate1_10B_P_233_F\" \"Plate1_10C_P_240_F\" \"Plate1_10D_P_244_F\" ...\n",
      "  .. .. .. .. .. .. ..@ x       : num [1:8385] 1.0908 -0.1274 1.1131 0.0236 -0.0815 ...\n",
      "  .. .. .. .. .. .. ..@ uplo    : chr \"U\"\n",
      "  .. .. .. .. .. .. ..@ factors : list()\n",
      "  .. .. .. .. ..$ p     : NULL\n",
      "  .. .. .. .. ..$ p_test: NULL\n",
      "  .. .. ..$ Qt                  : NULL\n",
      "  .. .. ..$ ZKZts               :List of 3\n",
      "  .. .. .. ..$ Family.(Intercept):Formal class 'dgCMatrix' [package \"Matrix\"] with 6 slots\n",
      "  .. .. .. .. .. ..@ i       : int [1:149769] 0 1 2 3 4 5 6 7 8 9 ...\n",
      "  .. .. .. .. .. ..@ p       : int [1:388] 0 387 774 1161 1548 1935 2322 2709 3096 3483 ...\n",
      "  .. .. .. .. .. ..@ Dim     : int [1:2] 387 387\n",
      "  .. .. .. .. .. ..@ Dimnames:List of 2\n",
      "  .. .. .. .. .. .. ..$ : chr [1:387] \"1\" \"2\" \"3\" \"4\" ...\n",
      "  .. .. .. .. .. .. ..$ : chr [1:387] \"1\" \"2\" \"3\" \"4\" ...\n",
      "  .. .. .. .. .. ..@ x       : num [1:149769] 1.007 1.007 1.007 0.0848 0.0848 ...\n",
      "  .. .. .. .. .. ..@ factors : list()\n",
      "  .. .. .. ..$ Family.TreatmentHS:Formal class 'dgCMatrix' [package \"Matrix\"] with 6 slots\n",
      "  .. .. .. .. .. ..@ i       : int [1:16641] 1 4 7 10 13 16 19 22 25 28 ...\n",
      "  .. .. .. .. .. ..@ p       : int [1:388] 0 0 129 129 129 258 258 258 387 387 ...\n",
      "  .. .. .. .. .. ..@ Dim     : int [1:2] 387 387\n",
      "  .. .. .. .. .. ..@ Dimnames:List of 2\n",
      "  .. .. .. .. .. .. ..$ : chr [1:387] \"1\" \"2\" \"3\" \"4\" ...\n",
      "  .. .. .. .. .. .. ..$ : chr [1:387] \"1\" \"2\" \"3\" \"4\" ...\n",
      "  .. .. .. .. .. ..@ x       : num [1:16641] 3.021 0.2543 -0.1691 -0.1904 0.0291 ...\n",
      "  .. .. .. .. .. ..@ factors : list()\n",
      "  .. .. .. ..$ Family.TreatmentR :Formal class 'dgCMatrix' [package \"Matrix\"] with 6 slots\n",
      "  .. .. .. .. .. ..@ i       : int [1:16641] 2 5 8 11 14 17 20 23 26 29 ...\n",
      "  .. .. .. .. .. ..@ p       : int [1:388] 0 0 0 129 129 129 258 258 258 387 ...\n",
      "  .. .. .. .. .. ..@ Dim     : int [1:2] 387 387\n",
      "  .. .. .. .. .. ..@ Dimnames:List of 2\n",
      "  .. .. .. .. .. .. ..$ : chr [1:387] \"1\" \"2\" \"3\" \"4\" ...\n",
      "  .. .. .. .. .. .. ..$ : chr [1:387] \"1\" \"2\" \"3\" \"4\" ...\n",
      "  .. .. .. .. .. ..@ x       : num [1:16641] 3.021 0.2543 -0.1691 -0.1904 0.0291 ...\n",
      "  .. .. .. .. .. ..@ factors : list()\n",
      "  .. .. ..$ resid_V             : num [1:387, 1:387] 1 0 0 0 0 0 0 0 0 0 ...\n",
      "  .. .. ..$ save_V_folder       : NULL\n",
      "  .. .. ..$ n_SNPs_downdated_RRM: Named num [1:3] Inf Inf Inf\n",
      "  .. .. .. ..- attr(*, \"names\")= chr [1:3] \"Family.(Intercept)\" \"Family.TreatmentHS\" \"Family.TreatmentR\"\n",
      "  .. .. ..$ downdate_ratios     : Named num [1:3] 1 1 1\n",
      "  .. .. .. ..- attr(*, \"names\")= chr [1:3] \"Family.(Intercept)\" \"Family.TreatmentHS\" \"Family.TreatmentR\"\n",
      "  .. ..$ mm_test         : num [1:387, 1:3] 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. .. ..$ : chr [1:387] \"1\" \"2\" \"3\" \"4\" ...\n",
      "  .. .. .. ..$ : chr [1:3] \"(Intercept)\" \"TreatmentHS\" \"TreatmentR\"\n",
      "  .. .. ..- attr(*, \"assign\")= int [1:3] 0 1 1\n",
      "  .. .. ..- attr(*, \"contrasts\")=List of 1\n",
      "  .. .. .. ..$ Treatment: chr \"contr.treatment\"\n",
      "  .. ..$ mm_reduced      : num [1:387, 1] 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. .. ..$ : chr [1:387] \"1\" \"2\" \"3\" \"4\" ...\n",
      "  .. .. .. ..$ : chr \"(Intercept)\"\n",
      "  .. .. ..- attr(*, \"assign\")= int 0\n",
      "  .. ..$ proximal_markers: NULL\n",
      "  .. ..$ proximal_Xs     : NULL\n",
      " $ :List of 2\n",
      "  ..$ results:'data.frame':\t565448 obs. of  25 variables:\n",
      "  .. ..$ Trait                  : chr [1:565448] \"y\" \"y\" \"y\" \"y\" ...\n",
      "  .. ..$ X_ID                   : chr [1:565448] \"2L:5095\" \"2L:5233\" \"2L:5372\" \"2L:5904\" ...\n",
      "  .. ..$ s2                     : num [1:565448] 7.33e+08 7.30e+08 7.29e+08 7.31e+08 7.30e+08 ...\n",
      "  .. ..$ ML_logLik              : num [1:565448] -4438 -4439 -4439 -4439 -4439 ...\n",
      "  .. ..$ Family..Intercept..ML  : num [1:565448] 0.518 0.508 0.508 0.508 0.508 ...\n",
      "  .. ..$ Family.TreatmentHS.ML  : num [1:565448] 0 0 0 0 0 0 0 0 0 0 ...\n",
      "  .. ..$ Family.TreatmentR.ML   : num [1:565448] 0 0 0 0 0 0 0 0 0 0 ...\n",
      "  .. ..$ beta.1                 : num [1:565448] -6782 -13106 -19270 -7658 -7554 ...\n",
      "  .. ..$ beta.2                 : num [1:565448] -13631 -11768 3713 -11824 -15136 ...\n",
      "  .. ..$ beta.3                 : num [1:565448] 1571 4882 2622 628 -246 ...\n",
      "  .. ..$ beta.4                 : num [1:565448] -1639 2612 5860 -555 -573 ...\n",
      "  .. ..$ beta.5                 : num [1:565448] 4389 1437 -6980 1871 3799 ...\n",
      "  .. ..$ beta.6                 : num [1:565448] -748 -2277 -976 134 680 ...\n",
      "  .. ..$ REML_logLik            : num [1:565448] -4372 -4374 -4373 -4374 -4374 ...\n",
      "  .. ..$ Family..Intercept..REML: num [1:565448] 0.518 0.508 0.508 0.508 0.508 ...\n",
      "  .. ..$ Family.TreatmentHS.REML: num [1:565448] 0 0 0 0 0 0 0 0 0 0 ...\n",
      "  .. ..$ Family.TreatmentR.REML : num [1:565448] 0 0 0 0 0 0 0 0 0 0 ...\n",
      "  .. ..$ F.1                    : num [1:565448] 0.3449 0.3014 0.7023 0.0234 0.0238 ...\n",
      "  .. ..$ F.2                    : num [1:565448] 2.3159 0.0897 0.964 0.2475 0.9945 ...\n",
      "  .. ..$ F.3                    : num [1:565448] 0.06732 0.22522 0.01886 0.00126 0.03183 ...\n",
      "  .. ..$ n_steps                : num [1:565448] 2 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ Df_X                   : int [1:565448] 3 3 3 3 3 3 3 3 3 3 ...\n",
      "  .. ..$ p_value_REML.1         : num [1:565448] 0.557 0.583 0.403 0.879 0.878 ...\n",
      "  .. ..$ p_value_REML.2         : num [1:565448] 0.129 0.765 0.327 0.619 0.319 ...\n",
      "  .. ..$ p_value_REML.3         : num [1:565448] 0.795 0.635 0.891 0.972 0.858 ...\n",
      "  ..$ setup  :List of 8\n",
      "  .. ..$ Y               : num [1:387, 1] -25591 -17786 -3112 69202 27184 ...\n",
      "  .. .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. .. ..$ : NULL\n",
      "  .. .. .. ..$ : chr \"y\"\n",
      "  .. ..$ X_cov           : num [1:387, 1:3] 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. .. ..$ : chr [1:387] \"1\" \"2\" \"3\" \"4\" ...\n",
      "  .. .. .. ..$ : chr [1:3] \"(Intercept)\" \"TreatmentHS\" \"TreatmentR\"\n",
      "  .. .. ..- attr(*, \"assign\")= int [1:3] 0 1 1\n",
      "  .. .. ..- attr(*, \"contrasts\")=List of 1\n",
      "  .. .. .. ..$ Treatment: chr \"contr.treatment\"\n",
      "  .. .. ..- attr(*, \"msgScaleX\")= chr(0) \n",
      "  .. ..$ h2_start        : num [1, 1:3] 0.508 0 0\n",
      "  .. .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. .. ..$ : NULL\n",
      "  .. .. .. ..$ : chr [1:3] \"Family..Intercept.\" \"Family.TreatmentHS\" \"Family.TreatmentR\"\n",
      "  .. ..$ V_setup         :List of 7\n",
      "  .. .. ..$ RE_setup            :List of 3\n",
      "  .. .. .. ..$ Family.(Intercept):List of 5\n",
      "  .. .. .. .. ..$ term  : chr \"Family\"\n",
      "  .. .. .. .. ..$ Z     :Formal class 'dgCMatrix' [package \"Matrix\"] with 6 slots\n",
      "  .. .. .. .. .. .. ..@ i       : int [1:387] 171 172 173 174 175 176 177 178 179 180 ...\n",
      "  .. .. .. .. .. .. ..@ p       : int [1:130] 0 3 6 9 12 15 18 21 24 27 ...\n",
      "  .. .. .. .. .. .. ..@ Dim     : int [1:2] 387 129\n",
      "  .. .. .. .. .. .. ..@ Dimnames:List of 2\n",
      "  .. .. .. .. .. .. .. ..$ : chr [1:387] \"1\" \"2\" \"3\" \"4\" ...\n",
      "  .. .. .. .. .. .. .. ..$ : chr [1:129] \"228\" \"233\" \"240\" \"244\" ...\n",
      "  .. .. .. .. .. .. ..@ x       : num [1:387] 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. .. .. .. .. .. ..@ factors : list()\n",
      "  .. .. .. .. ..$ K     :Formal class 'dsCMatrix' [package \"Matrix\"] with 7 slots\n",
      "  .. .. .. .. .. .. ..@ i       : int [1:8385] 0 0 1 0 1 2 0 1 2 3 ...\n",
      "  .. .. .. .. .. .. ..@ p       : int [1:130] 0 1 3 6 10 15 21 28 36 45 ...\n",
      "  .. .. .. .. .. .. ..@ Dim     : int [1:2] 129 129\n",
      "  .. .. .. .. .. .. ..@ Dimnames:List of 2\n",
      "  .. .. .. .. .. .. .. ..$ : Named chr [1:129] \"228\" \"233\" \"240\" \"244\" ...\n",
      "  .. .. .. .. .. .. .. .. ..- attr(*, \"names\")= chr [1:129] \"Plate1_10A_P_228_F\" \"Plate1_10B_P_233_F\" \"Plate1_10C_P_240_F\" \"Plate1_10D_P_244_F\" ...\n",
      "  .. .. .. .. .. .. .. ..$ : Named chr [1:129] \"228\" \"233\" \"240\" \"244\" ...\n",
      "  .. .. .. .. .. .. .. .. ..- attr(*, \"names\")= chr [1:129] \"Plate1_10A_P_228_F\" \"Plate1_10B_P_233_F\" \"Plate1_10C_P_240_F\" \"Plate1_10D_P_244_F\" ...\n",
      "  .. .. .. .. .. .. ..@ x       : num [1:8385] 1.0908 -0.1274 1.1131 0.0236 -0.0815 ...\n",
      "  .. .. .. .. .. .. ..@ uplo    : chr \"U\"\n",
      "  .. .. .. .. .. .. ..@ factors : list()\n",
      "  .. .. .. .. ..$ p     : NULL\n",
      "  .. .. .. .. ..$ p_test: NULL\n",
      "  .. .. .. ..$ Family.TreatmentHS:List of 5\n",
      "  .. .. .. .. ..$ term  : chr \"Family\"\n",
      "  .. .. .. .. ..$ Z     :Formal class 'dgCMatrix' [package \"Matrix\"] with 6 slots\n",
      "  .. .. .. .. .. .. ..@ i       : int [1:129] 172 175 178 181 184 187 190 193 196 199 ...\n",
      "  .. .. .. .. .. .. ..@ p       : int [1:130] 0 1 2 3 4 5 6 7 8 9 ...\n",
      "  .. .. .. .. .. .. ..@ Dim     : int [1:2] 387 129\n",
      "  .. .. .. .. .. .. ..@ Dimnames:List of 2\n",
      "  .. .. .. .. .. .. .. ..$ : chr [1:387] \"1\" \"2\" \"3\" \"4\" ...\n",
      "  .. .. .. .. .. .. .. ..$ : chr [1:129] \"228\" \"233\" \"240\" \"244\" ...\n",
      "  .. .. .. .. .. .. ..@ x       : num [1:129] 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. .. .. .. .. .. ..@ factors : list()\n",
      "  .. .. .. .. ..$ K     :Formal class 'dsCMatrix' [package \"Matrix\"] with 7 slots\n",
      "  .. .. .. .. .. .. ..@ i       : int [1:8385] 0 0 1 0 1 2 0 1 2 3 ...\n",
      "  .. .. .. .. .. .. ..@ p       : int [1:130] 0 1 3 6 10 15 21 28 36 45 ...\n",
      "  .. .. .. .. .. .. ..@ Dim     : int [1:2] 129 129\n",
      "  .. .. .. .. .. .. ..@ Dimnames:List of 2\n",
      "  .. .. .. .. .. .. .. ..$ : Named chr [1:129] \"228\" \"233\" \"240\" \"244\" ...\n",
      "  .. .. .. .. .. .. .. .. ..- attr(*, \"names\")= chr [1:129] \"Plate1_10A_P_228_F\" \"Plate1_10B_P_233_F\" \"Plate1_10C_P_240_F\" \"Plate1_10D_P_244_F\" ...\n",
      "  .. .. .. .. .. .. .. ..$ : Named chr [1:129] \"228\" \"233\" \"240\" \"244\" ...\n",
      "  .. .. .. .. .. .. .. .. ..- attr(*, \"names\")= chr [1:129] \"Plate1_10A_P_228_F\" \"Plate1_10B_P_233_F\" \"Plate1_10C_P_240_F\" \"Plate1_10D_P_244_F\" ...\n",
      "  .. .. .. .. .. .. ..@ x       : num [1:8385] 1.0908 -0.1274 1.1131 0.0236 -0.0815 ...\n",
      "  .. .. .. .. .. .. ..@ uplo    : chr \"U\"\n",
      "  .. .. .. .. .. .. ..@ factors : list()\n",
      "  .. .. .. .. ..$ p     : NULL\n",
      "  .. .. .. .. ..$ p_test: NULL\n",
      "  .. .. .. ..$ Family.TreatmentR :List of 5\n",
      "  .. .. .. .. ..$ term  : chr \"Family\"\n",
      "  .. .. .. .. ..$ Z     :Formal class 'dgCMatrix' [package \"Matrix\"] with 6 slots\n",
      "  .. .. .. .. .. .. ..@ i       : int [1:129] 173 176 179 182 185 188 191 194 197 200 ...\n",
      "  .. .. .. .. .. .. ..@ p       : int [1:130] 0 1 2 3 4 5 6 7 8 9 ...\n",
      "  .. .. .. .. .. .. ..@ Dim     : int [1:2] 387 129\n",
      "  .. .. .. .. .. .. ..@ Dimnames:List of 2\n",
      "  .. .. .. .. .. .. .. ..$ : chr [1:387] \"1\" \"2\" \"3\" \"4\" ...\n",
      "  .. .. .. .. .. .. .. ..$ : chr [1:129] \"228\" \"233\" \"240\" \"244\" ...\n",
      "  .. .. .. .. .. .. ..@ x       : num [1:129] 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. .. .. .. .. .. ..@ factors : list()\n",
      "  .. .. .. .. ..$ K     :Formal class 'dsCMatrix' [package \"Matrix\"] with 7 slots\n",
      "  .. .. .. .. .. .. ..@ i       : int [1:8385] 0 0 1 0 1 2 0 1 2 3 ...\n",
      "  .. .. .. .. .. .. ..@ p       : int [1:130] 0 1 3 6 10 15 21 28 36 45 ...\n",
      "  .. .. .. .. .. .. ..@ Dim     : int [1:2] 129 129\n",
      "  .. .. .. .. .. .. ..@ Dimnames:List of 2\n",
      "  .. .. .. .. .. .. .. ..$ : Named chr [1:129] \"228\" \"233\" \"240\" \"244\" ...\n",
      "  .. .. .. .. .. .. .. .. ..- attr(*, \"names\")= chr [1:129] \"Plate1_10A_P_228_F\" \"Plate1_10B_P_233_F\" \"Plate1_10C_P_240_F\" \"Plate1_10D_P_244_F\" ...\n",
      "  .. .. .. .. .. .. .. ..$ : Named chr [1:129] \"228\" \"233\" \"240\" \"244\" ...\n",
      "  .. .. .. .. .. .. .. .. ..- attr(*, \"names\")= chr [1:129] \"Plate1_10A_P_228_F\" \"Plate1_10B_P_233_F\" \"Plate1_10C_P_240_F\" \"Plate1_10D_P_244_F\" ...\n",
      "  .. .. .. .. .. .. ..@ x       : num [1:8385] 1.0908 -0.1274 1.1131 0.0236 -0.0815 ...\n",
      "  .. .. .. .. .. .. ..@ uplo    : chr \"U\"\n",
      "  .. .. .. .. .. .. ..@ factors : list()\n",
      "  .. .. .. .. ..$ p     : NULL\n",
      "  .. .. .. .. ..$ p_test: NULL\n",
      "  .. .. ..$ Qt                  : NULL\n",
      "  .. .. ..$ ZKZts               :List of 3\n",
      "  .. .. .. ..$ Family.(Intercept):Formal class 'dgCMatrix' [package \"Matrix\"] with 6 slots\n",
      "  .. .. .. .. .. ..@ i       : int [1:149769] 0 1 2 3 4 5 6 7 8 9 ...\n",
      "  .. .. .. .. .. ..@ p       : int [1:388] 0 387 774 1161 1548 1935 2322 2709 3096 3483 ...\n",
      "  .. .. .. .. .. ..@ Dim     : int [1:2] 387 387\n",
      "  .. .. .. .. .. ..@ Dimnames:List of 2\n",
      "  .. .. .. .. .. .. ..$ : chr [1:387] \"1\" \"2\" \"3\" \"4\" ...\n",
      "  .. .. .. .. .. .. ..$ : chr [1:387] \"1\" \"2\" \"3\" \"4\" ...\n",
      "  .. .. .. .. .. ..@ x       : num [1:149769] 1.007 1.007 1.007 0.0848 0.0848 ...\n",
      "  .. .. .. .. .. ..@ factors : list()\n",
      "  .. .. .. ..$ Family.TreatmentHS:Formal class 'dgCMatrix' [package \"Matrix\"] with 6 slots\n",
      "  .. .. .. .. .. ..@ i       : int [1:16641] 1 4 7 10 13 16 19 22 25 28 ...\n",
      "  .. .. .. .. .. ..@ p       : int [1:388] 0 0 129 129 129 258 258 258 387 387 ...\n",
      "  .. .. .. .. .. ..@ Dim     : int [1:2] 387 387\n",
      "  .. .. .. .. .. ..@ Dimnames:List of 2\n",
      "  .. .. .. .. .. .. ..$ : chr [1:387] \"1\" \"2\" \"3\" \"4\" ...\n",
      "  .. .. .. .. .. .. ..$ : chr [1:387] \"1\" \"2\" \"3\" \"4\" ...\n",
      "  .. .. .. .. .. ..@ x       : num [1:16641] 3.021 0.2543 -0.1691 -0.1904 0.0291 ...\n",
      "  .. .. .. .. .. ..@ factors : list()\n",
      "  .. .. .. ..$ Family.TreatmentR :Formal class 'dgCMatrix' [package \"Matrix\"] with 6 slots\n",
      "  .. .. .. .. .. ..@ i       : int [1:16641] 2 5 8 11 14 17 20 23 26 29 ...\n",
      "  .. .. .. .. .. ..@ p       : int [1:388] 0 0 0 129 129 129 258 258 258 387 ...\n",
      "  .. .. .. .. .. ..@ Dim     : int [1:2] 387 387\n",
      "  .. .. .. .. .. ..@ Dimnames:List of 2\n",
      "  .. .. .. .. .. .. ..$ : chr [1:387] \"1\" \"2\" \"3\" \"4\" ...\n",
      "  .. .. .. .. .. .. ..$ : chr [1:387] \"1\" \"2\" \"3\" \"4\" ...\n",
      "  .. .. .. .. .. ..@ x       : num [1:16641] 3.021 0.2543 -0.1691 -0.1904 0.0291 ...\n",
      "  .. .. .. .. .. ..@ factors : list()\n",
      "  .. .. ..$ resid_V             : num [1:387, 1:387] 1 0 0 0 0 0 0 0 0 0 ...\n",
      "  .. .. ..$ save_V_folder       : NULL\n",
      "  .. .. ..$ n_SNPs_downdated_RRM: Named num [1:3] Inf Inf Inf\n",
      "  .. .. .. ..- attr(*, \"names\")= chr [1:3] \"Family.(Intercept)\" \"Family.TreatmentHS\" \"Family.TreatmentR\"\n",
      "  .. .. ..$ downdate_ratios     : Named num [1:3] 1 1 1\n",
      "  .. .. .. ..- attr(*, \"names\")= chr [1:3] \"Family.(Intercept)\" \"Family.TreatmentHS\" \"Family.TreatmentR\"\n",
      "  .. ..$ mm_test         : num [1:387, 1:3] 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. .. ..$ : chr [1:387] \"1\" \"2\" \"3\" \"4\" ...\n",
      "  .. .. .. ..$ : chr [1:3] \"(Intercept)\" \"TreatmentHS\" \"TreatmentR\"\n",
      "  .. .. ..- attr(*, \"assign\")= int [1:3] 0 1 1\n",
      "  .. .. ..- attr(*, \"contrasts\")=List of 1\n",
      "  .. .. .. ..$ Treatment: chr \"contr.treatment\"\n",
      "  .. ..$ mm_reduced      : num [1:387, 1] 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. .. ..- attr(*, \"dimnames\")=List of 2\n",
      "  .. .. .. ..$ : chr [1:387] \"1\" \"2\" \"3\" \"4\" ...\n",
      "  .. .. .. ..$ : chr \"(Intercept)\"\n",
      "  .. .. ..- attr(*, \"assign\")= int 0\n",
      "  .. ..$ proximal_markers: NULL\n",
      "  .. ..$ proximal_Xs     : NULL\n"
     ]
    }
   ],
   "source": [
    "str(out_32to36_grm_regressed)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "b3cd4f3d",
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in library(ggman): there is no package called ‘ggman’\n",
     "output_type": "error",
     "traceback": [
      "Error in library(ggman): there is no package called ‘ggman’\nTraceback:\n",
      "1. library(ggman)"
     ]
    }
   ],
   "source": [
    "library(ggman)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.0.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
